{
 "cells": [
  {
   "cell_type": "code",
   "id": "initial_id",
   "metadata": {
    "collapsed": true,
    "ExecuteTime": {
     "end_time": "2024-11-07T19:52:53.703844Z",
     "start_time": "2024-11-07T19:52:53.476261Z"
    }
   },
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "import pandas as pd\n",
    "import patternq.query as pqq\n",
    "import patternq.dataset as pqd\n",
    "import patternq.reference as pqr"
   ],
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/benjaminkamphaus/Library/Caches/pypoetry/virtualenvs/patternq-bsnIQB4Y-py3.9/lib/python3.9/site-packages/urllib3/__init__.py:35: NotOpenSSLWarning: urllib3 v2 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with 'LibreSSL 2.8.3'. See: https://github.com/urllib3/urllib3/issues/3020\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "execution_count": 1
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-11-07T19:58:38.091191Z",
     "start_time": "2024-11-07T19:58:36.111239Z"
    }
   },
   "cell_type": "code",
   "source": [
    "subjects = pqd.all_subjects('tcga-brca', db_name='tcga-brca')\n",
    "subjects"
   ],
   "id": "79f2439641b3d0d6",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                          subject-HLA-A-type  \\\n",
       "0                           [hla_a_68_01_02]   \n",
       "1           [hla_a_25_01_01, hla_a_31_01_02]   \n",
       "2                        [hla_a_03_01_01_01]   \n",
       "3                           [hla_a_26_01_03]   \n",
       "4                        [hla_a_02_01_01_01]   \n",
       "...                                      ...   \n",
       "1093     [hla_a_01_01_01_01, hla_a_26_01_01]   \n",
       "1094  [hla_a_01_01_01_01, hla_a_24_02_01_01]   \n",
       "1095                     [hla_a_02_01_01_01]   \n",
       "1096     [hla_a_31_01_02, hla_a_80_01_01_01]   \n",
       "1097                                     NaN   \n",
       "\n",
       "                                        subject-uid  \\\n",
       "0     [tcga-brca, 48958e0a033b76dd2cde746ec340d7b8]   \n",
       "1     [tcga-brca, cdff6ccae7b3a01e06583b6eb6935647]   \n",
       "2     [tcga-brca, 58de8decc5806306e122d7d2e0361077]   \n",
       "3     [tcga-brca, e33c10ac862066f01799593f93226d63]   \n",
       "4     [tcga-brca, 33c8d0592a63e6d929493c6fee83049a]   \n",
       "...                                             ...   \n",
       "1093  [tcga-brca, 28355fea1611cf76788e36f7d8d1e63b]   \n",
       "1094  [tcga-brca, b49d76c1ed3ce2f807c30aca593700e5]   \n",
       "1095  [tcga-brca, 679b389e478544f624e010f2fba4951c]   \n",
       "1096  [tcga-brca, ac707e57a213016c5f7f1a643c938bc0]   \n",
       "1097  [tcga-brca, bd92ea29e8454b46c4e0c940f391887a]   \n",
       "\n",
       "                        subject-race   subject-freetext-disease  \\\n",
       "0                :subject.race/white  Invasive breast carcinoma   \n",
       "1                :subject.race/white  Invasive breast carcinoma   \n",
       "2                :subject.race/white  Invasive breast carcinoma   \n",
       "3                :subject.race/white  Invasive breast carcinoma   \n",
       "4                :subject.race/white  Invasive breast carcinoma   \n",
       "...                              ...                        ...   \n",
       "1093             :subject.race/white  Invasive breast carcinoma   \n",
       "1094                             NaN  Invasive breast carcinoma   \n",
       "1095             :subject.race/white  Invasive breast carcinoma   \n",
       "1096  :subject.race/african-american  Invasive breast carcinoma   \n",
       "1097                             NaN  Invasive breast carcinoma   \n",
       "\n",
       "                       subject-HLA-B-type           db-id  subject-age  \\\n",
       "0     [hla_b_40_01_02, hla_b_44_02_01_03]  17592187509997         50.0   \n",
       "1        [hla_b_18_01_01_02, hla_b_51_08]  17592187510001         68.0   \n",
       "2                        [hla_b_07_02_01]  17592187510005         35.0   \n",
       "3        [hla_b_38_01_01, hla_b_40_02_01]  17592187510009         55.0   \n",
       "4     [hla_b_13_02_01, hla_b_18_01_01_02]  17592187510013         83.0   \n",
       "...                                   ...             ...          ...   \n",
       "1093     [hla_b_07_02_01, hla_b_08_01_07]  17592187514126         44.0   \n",
       "1094     [hla_b_08_01_01, hla_b_44_05_01]  17592187514130         63.0   \n",
       "1095  [hla_b_07_02_01, hla_b_35_01_01_02]  17592187514133         53.0   \n",
       "1096     [hla_b_27_05_02, hla_b_40_01_02]  17592187517522          NaN   \n",
       "1097                                  NaN  17592234961832          NaN   \n",
       "\n",
       "              subject-sex                      subject-HLA-C-type  \\\n",
       "0     :subject.sex/female     [hla_c_03_04_01_01, hla_c_07_04_01]   \n",
       "1     :subject.sex/female     [hla_c_12_03_01_01, hla_c_16_02_01]   \n",
       "2     :subject.sex/female                     [hla_c_07_02_01_03]   \n",
       "3     :subject.sex/female     [hla_c_02_02_02, hla_c_12_03_01_01]   \n",
       "4     :subject.sex/female  [hla_c_06_02_01_01, hla_c_07_01_01_01]   \n",
       "...                   ...                                     ...   \n",
       "1093  :subject.sex/female  [hla_c_07_01_01_01, hla_c_07_02_01_03]   \n",
       "1094  :subject.sex/female     [hla_c_02_02_02, hla_c_07_01_01_01]   \n",
       "1095  :subject.sex/female  [hla_c_04_01_01_01, hla_c_07_02_01_03]   \n",
       "1096  :subject.sex/female  [hla_c_03_04_01_01, hla_c_06_02_01_01]   \n",
       "1097                  NaN                                     NaN   \n",
       "\n",
       "     subject-dead    subject-id                subject-ethnicity  \n",
       "0            True  TCGA-AR-A1AR                              NaN  \n",
       "1            True  TCGA-BH-A1EO                              NaN  \n",
       "2            True  TCGA-BH-A1ES                              NaN  \n",
       "3            True  TCGA-BH-A1ET                              NaN  \n",
       "4            True  TCGA-BH-A1EU                              NaN  \n",
       "...           ...           ...                              ...  \n",
       "1093        False  TCGA-E2-A1B6                              NaN  \n",
       "1094        False  TCGA-E2-A1BC                              NaN  \n",
       "1095        False  TCGA-E2-A1BD                              NaN  \n",
       "1096        False  TCGA-OL-A66H  :subject.ethnicity/not-hispanic  \n",
       "1097          NaN  TCGA-BH-A0B2                              NaN  \n",
       "\n",
       "[1098 rows x 12 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject-HLA-A-type</th>\n",
       "      <th>subject-uid</th>\n",
       "      <th>subject-race</th>\n",
       "      <th>subject-freetext-disease</th>\n",
       "      <th>subject-HLA-B-type</th>\n",
       "      <th>db-id</th>\n",
       "      <th>subject-age</th>\n",
       "      <th>subject-sex</th>\n",
       "      <th>subject-HLA-C-type</th>\n",
       "      <th>subject-dead</th>\n",
       "      <th>subject-id</th>\n",
       "      <th>subject-ethnicity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>[hla_a_68_01_02]</td>\n",
       "      <td>[tcga-brca, 48958e0a033b76dd2cde746ec340d7b8]</td>\n",
       "      <td>:subject.race/white</td>\n",
       "      <td>Invasive breast carcinoma</td>\n",
       "      <td>[hla_b_40_01_02, hla_b_44_02_01_03]</td>\n",
       "      <td>17592187509997</td>\n",
       "      <td>50.0</td>\n",
       "      <td>:subject.sex/female</td>\n",
       "      <td>[hla_c_03_04_01_01, hla_c_07_04_01]</td>\n",
       "      <td>True</td>\n",
       "      <td>TCGA-AR-A1AR</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>[hla_a_25_01_01, hla_a_31_01_02]</td>\n",
       "      <td>[tcga-brca, cdff6ccae7b3a01e06583b6eb6935647]</td>\n",
       "      <td>:subject.race/white</td>\n",
       "      <td>Invasive breast carcinoma</td>\n",
       "      <td>[hla_b_18_01_01_02, hla_b_51_08]</td>\n",
       "      <td>17592187510001</td>\n",
       "      <td>68.0</td>\n",
       "      <td>:subject.sex/female</td>\n",
       "      <td>[hla_c_12_03_01_01, hla_c_16_02_01]</td>\n",
       "      <td>True</td>\n",
       "      <td>TCGA-BH-A1EO</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>[hla_a_03_01_01_01]</td>\n",
       "      <td>[tcga-brca, 58de8decc5806306e122d7d2e0361077]</td>\n",
       "      <td>:subject.race/white</td>\n",
       "      <td>Invasive breast carcinoma</td>\n",
       "      <td>[hla_b_07_02_01]</td>\n",
       "      <td>17592187510005</td>\n",
       "      <td>35.0</td>\n",
       "      <td>:subject.sex/female</td>\n",
       "      <td>[hla_c_07_02_01_03]</td>\n",
       "      <td>True</td>\n",
       "      <td>TCGA-BH-A1ES</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>[hla_a_26_01_03]</td>\n",
       "      <td>[tcga-brca, e33c10ac862066f01799593f93226d63]</td>\n",
       "      <td>:subject.race/white</td>\n",
       "      <td>Invasive breast carcinoma</td>\n",
       "      <td>[hla_b_38_01_01, hla_b_40_02_01]</td>\n",
       "      <td>17592187510009</td>\n",
       "      <td>55.0</td>\n",
       "      <td>:subject.sex/female</td>\n",
       "      <td>[hla_c_02_02_02, hla_c_12_03_01_01]</td>\n",
       "      <td>True</td>\n",
       "      <td>TCGA-BH-A1ET</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>[hla_a_02_01_01_01]</td>\n",
       "      <td>[tcga-brca, 33c8d0592a63e6d929493c6fee83049a]</td>\n",
       "      <td>:subject.race/white</td>\n",
       "      <td>Invasive breast carcinoma</td>\n",
       "      <td>[hla_b_13_02_01, hla_b_18_01_01_02]</td>\n",
       "      <td>17592187510013</td>\n",
       "      <td>83.0</td>\n",
       "      <td>:subject.sex/female</td>\n",
       "      <td>[hla_c_06_02_01_01, hla_c_07_01_01_01]</td>\n",
       "      <td>True</td>\n",
       "      <td>TCGA-BH-A1EU</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1093</th>\n",
       "      <td>[hla_a_01_01_01_01, hla_a_26_01_01]</td>\n",
       "      <td>[tcga-brca, 28355fea1611cf76788e36f7d8d1e63b]</td>\n",
       "      <td>:subject.race/white</td>\n",
       "      <td>Invasive breast carcinoma</td>\n",
       "      <td>[hla_b_07_02_01, hla_b_08_01_07]</td>\n",
       "      <td>17592187514126</td>\n",
       "      <td>44.0</td>\n",
       "      <td>:subject.sex/female</td>\n",
       "      <td>[hla_c_07_01_01_01, hla_c_07_02_01_03]</td>\n",
       "      <td>False</td>\n",
       "      <td>TCGA-E2-A1B6</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1094</th>\n",
       "      <td>[hla_a_01_01_01_01, hla_a_24_02_01_01]</td>\n",
       "      <td>[tcga-brca, b49d76c1ed3ce2f807c30aca593700e5]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Invasive breast carcinoma</td>\n",
       "      <td>[hla_b_08_01_01, hla_b_44_05_01]</td>\n",
       "      <td>17592187514130</td>\n",
       "      <td>63.0</td>\n",
       "      <td>:subject.sex/female</td>\n",
       "      <td>[hla_c_02_02_02, hla_c_07_01_01_01]</td>\n",
       "      <td>False</td>\n",
       "      <td>TCGA-E2-A1BC</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1095</th>\n",
       "      <td>[hla_a_02_01_01_01]</td>\n",
       "      <td>[tcga-brca, 679b389e478544f624e010f2fba4951c]</td>\n",
       "      <td>:subject.race/white</td>\n",
       "      <td>Invasive breast carcinoma</td>\n",
       "      <td>[hla_b_07_02_01, hla_b_35_01_01_02]</td>\n",
       "      <td>17592187514133</td>\n",
       "      <td>53.0</td>\n",
       "      <td>:subject.sex/female</td>\n",
       "      <td>[hla_c_04_01_01_01, hla_c_07_02_01_03]</td>\n",
       "      <td>False</td>\n",
       "      <td>TCGA-E2-A1BD</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>[hla_a_31_01_02, hla_a_80_01_01_01]</td>\n",
       "      <td>[tcga-brca, ac707e57a213016c5f7f1a643c938bc0]</td>\n",
       "      <td>:subject.race/african-american</td>\n",
       "      <td>Invasive breast carcinoma</td>\n",
       "      <td>[hla_b_27_05_02, hla_b_40_01_02]</td>\n",
       "      <td>17592187517522</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:subject.sex/female</td>\n",
       "      <td>[hla_c_03_04_01_01, hla_c_06_02_01_01]</td>\n",
       "      <td>False</td>\n",
       "      <td>TCGA-OL-A66H</td>\n",
       "      <td>:subject.ethnicity/not-hispanic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>NaN</td>\n",
       "      <td>[tcga-brca, bd92ea29e8454b46c4e0c940f391887a]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Invasive breast carcinoma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17592234961832</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-BH-A0B2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1098 rows × 12 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 10
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-11-07T20:04:26.970181Z",
     "start_time": "2024-11-07T20:04:26.942163Z"
    }
   },
   "cell_type": "code",
   "source": [
    "by_hla = subjects[['subject-id', 'subject-HLA-A-type', 'subject-HLA-B-type', 'subject-HLA-C-type']].explode(column='subject-HLA-A-type')\n",
    "by_hla = by_hla.explode(column='subject-HLA-B-type')\n",
    "by_hla = by_hla.explode(column='subject-HLA-C-type')\n",
    "by_hla"
   ],
   "id": "eefb2df20c58950c",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        subject-id subject-HLA-A-type subject-HLA-B-type subject-HLA-C-type\n",
       "0     TCGA-AR-A1AR     hla_a_68_01_02     hla_b_40_01_02  hla_c_03_04_01_01\n",
       "0     TCGA-AR-A1AR     hla_a_68_01_02     hla_b_40_01_02     hla_c_07_04_01\n",
       "0     TCGA-AR-A1AR     hla_a_68_01_02  hla_b_44_02_01_03  hla_c_03_04_01_01\n",
       "0     TCGA-AR-A1AR     hla_a_68_01_02  hla_b_44_02_01_03     hla_c_07_04_01\n",
       "1     TCGA-BH-A1EO     hla_a_25_01_01  hla_b_18_01_01_02  hla_c_12_03_01_01\n",
       "...            ...                ...                ...                ...\n",
       "1096  TCGA-OL-A66H  hla_a_80_01_01_01     hla_b_27_05_02  hla_c_03_04_01_01\n",
       "1096  TCGA-OL-A66H  hla_a_80_01_01_01     hla_b_27_05_02  hla_c_06_02_01_01\n",
       "1096  TCGA-OL-A66H  hla_a_80_01_01_01     hla_b_40_01_02  hla_c_03_04_01_01\n",
       "1096  TCGA-OL-A66H  hla_a_80_01_01_01     hla_b_40_01_02  hla_c_06_02_01_01\n",
       "1097  TCGA-BH-A0B2                NaN                NaN                NaN\n",
       "\n",
       "[7217 rows x 4 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject-id</th>\n",
       "      <th>subject-HLA-A-type</th>\n",
       "      <th>subject-HLA-B-type</th>\n",
       "      <th>subject-HLA-C-type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-AR-A1AR</td>\n",
       "      <td>hla_a_68_01_02</td>\n",
       "      <td>hla_b_40_01_02</td>\n",
       "      <td>hla_c_03_04_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-AR-A1AR</td>\n",
       "      <td>hla_a_68_01_02</td>\n",
       "      <td>hla_b_40_01_02</td>\n",
       "      <td>hla_c_07_04_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-AR-A1AR</td>\n",
       "      <td>hla_a_68_01_02</td>\n",
       "      <td>hla_b_44_02_01_03</td>\n",
       "      <td>hla_c_03_04_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-AR-A1AR</td>\n",
       "      <td>hla_a_68_01_02</td>\n",
       "      <td>hla_b_44_02_01_03</td>\n",
       "      <td>hla_c_07_04_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-BH-A1EO</td>\n",
       "      <td>hla_a_25_01_01</td>\n",
       "      <td>hla_b_18_01_01_02</td>\n",
       "      <td>hla_c_12_03_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>TCGA-OL-A66H</td>\n",
       "      <td>hla_a_80_01_01_01</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_03_04_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>TCGA-OL-A66H</td>\n",
       "      <td>hla_a_80_01_01_01</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_06_02_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>TCGA-OL-A66H</td>\n",
       "      <td>hla_a_80_01_01_01</td>\n",
       "      <td>hla_b_40_01_02</td>\n",
       "      <td>hla_c_03_04_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>TCGA-OL-A66H</td>\n",
       "      <td>hla_a_80_01_01_01</td>\n",
       "      <td>hla_b_40_01_02</td>\n",
       "      <td>hla_c_06_02_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>TCGA-BH-A0B2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7217 rows × 4 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 18
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-11-07T20:09:56.927223Z",
     "start_time": "2024-11-07T20:09:56.908913Z"
    }
   },
   "cell_type": "code",
   "source": "by_hla['subject-HLA-B-type'].value_counts()",
   "id": "954d82c70875386b",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "subject-HLA-B-type\n",
       "hla_b_07_02_01       841\n",
       "hla_b_08_01_01       508\n",
       "hla_b_15_01_01_01    423\n",
       "hla_b_44_02_01_01    393\n",
       "hla_b_35_01_01_02    332\n",
       "                    ... \n",
       "hla_b_51_07_02         2\n",
       "hla_b_39_01_03         1\n",
       "hla_b_44_02_15         1\n",
       "hla_b_14_03            1\n",
       "hla_b_07_02_06         1\n",
       "Name: count, Length: 102, dtype: int64"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 27
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-11-07T20:12:46.796542Z",
     "start_time": "2024-11-07T20:12:46.760618Z"
    }
   },
   "cell_type": "code",
   "source": [
    "b27_matches = by_hla['subject-HLA-B-type'].str.contains(\"b_27\")\n",
    "b27_matches = b27_matches.fillna(False)\n",
    "b27_matches = b27_matches == True\n",
    "b27_sub = by_hla[b27_matches]\n",
    "b27_sub"
   ],
   "id": "df40656df2a27b62",
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/ym/gv3tr8lj5g3773_33yn9gr200000gn/T/ipykernel_60899/4069950816.py:2: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`\n",
      "  b27_matches = b27_matches.fillna(False)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "        subject-id subject-HLA-A-type subject-HLA-B-type subject-HLA-C-type\n",
       "15    TCGA-E2-A15I     hla_a_03_01_19     hla_b_27_05_02     hla_c_01_02_01\n",
       "15    TCGA-E2-A15I     hla_a_03_01_19     hla_b_27_05_02     hla_c_15_02_01\n",
       "15    TCGA-E2-A15I     hla_a_26_01_03     hla_b_27_05_02     hla_c_01_02_01\n",
       "15    TCGA-E2-A15I     hla_a_26_01_03     hla_b_27_05_02     hla_c_15_02_01\n",
       "33    TCGA-B6-A1KN  hla_a_03_01_01_01     hla_b_27_05_02     hla_c_02_02_02\n",
       "...            ...                ...                ...                ...\n",
       "1064  TCGA-AR-A1AQ     hla_a_02_06_05     hla_b_27_02_01     hla_c_16_01_01\n",
       "1096  TCGA-OL-A66H     hla_a_31_01_02     hla_b_27_05_02  hla_c_03_04_01_01\n",
       "1096  TCGA-OL-A66H     hla_a_31_01_02     hla_b_27_05_02  hla_c_06_02_01_01\n",
       "1096  TCGA-OL-A66H  hla_a_80_01_01_01     hla_b_27_05_02  hla_c_03_04_01_01\n",
       "1096  TCGA-OL-A66H  hla_a_80_01_01_01     hla_b_27_05_02  hla_c_06_02_01_01\n",
       "\n",
       "[297 rows x 4 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject-id</th>\n",
       "      <th>subject-HLA-A-type</th>\n",
       "      <th>subject-HLA-B-type</th>\n",
       "      <th>subject-HLA-C-type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TCGA-E2-A15I</td>\n",
       "      <td>hla_a_03_01_19</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_01_02_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TCGA-E2-A15I</td>\n",
       "      <td>hla_a_03_01_19</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_15_02_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TCGA-E2-A15I</td>\n",
       "      <td>hla_a_26_01_03</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_01_02_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TCGA-E2-A15I</td>\n",
       "      <td>hla_a_26_01_03</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_15_02_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>TCGA-B6-A1KN</td>\n",
       "      <td>hla_a_03_01_01_01</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_02_02_02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1064</th>\n",
       "      <td>TCGA-AR-A1AQ</td>\n",
       "      <td>hla_a_02_06_05</td>\n",
       "      <td>hla_b_27_02_01</td>\n",
       "      <td>hla_c_16_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>TCGA-OL-A66H</td>\n",
       "      <td>hla_a_31_01_02</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_03_04_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>TCGA-OL-A66H</td>\n",
       "      <td>hla_a_31_01_02</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_06_02_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>TCGA-OL-A66H</td>\n",
       "      <td>hla_a_80_01_01_01</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_03_04_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>TCGA-OL-A66H</td>\n",
       "      <td>hla_a_80_01_01_01</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_06_02_01_01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>297 rows × 4 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 33
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-11-07T20:13:22.637913Z",
     "start_time": "2024-11-07T20:13:11.934710Z"
    }
   },
   "cell_type": "code",
   "source": [
    "sub_id = \"TCGA-B6-A1KN\"\n",
    "assays_for_sub = pqd.assays_for_patient(sub_id, db_name='tcga-brca')\n",
    "assays_for_sub"
   ],
   "id": "bb87ca446c002c8a",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "     subject-id         sample-id                       assay-tech  \\\n",
       "0  TCGA-B6-A1KN  tcga-b6-a1kn-01a        :assay.technology/RNA-seq   \n",
       "1  TCGA-B6-A1KN  tcga-b6-a1kn-01a      :assay.technology/SNP-array   \n",
       "2  TCGA-B6-A1KN  tcga-b6-a1kn-01a            :assay.technology/WES   \n",
       "3  TCGA-B6-A1KN  tcga-b6-a1kn-10a        :assay.technology/RNA-seq   \n",
       "4  TCGA-B6-A1KN  tcga-b6-a1kn-01a        :assay.technology/RNA-seq   \n",
       "5  TCGA-B6-A1KN  tcga-b6-a1kn-01a  :assay.technology/miscellaneous   \n",
       "6  TCGA-B6-A1KN  tcga-b6-a1kn-10a      :assay.technology/SNP-array   \n",
       "\n",
       "         assay-name       measurement-set-name  \n",
       "0      RNA-seq data                         gx  \n",
       "1  cnv and variants               snp CNV data  \n",
       "2               WES         baseline mutations  \n",
       "3      RNA-seq data  immune-cell-deconvolution  \n",
       "4      RNA-seq data  immune-cell-deconvolution  \n",
       "5      Tumor purity               tumor purity  \n",
       "6  cnv and variants               snp CNV data  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject-id</th>\n",
       "      <th>sample-id</th>\n",
       "      <th>assay-tech</th>\n",
       "      <th>assay-name</th>\n",
       "      <th>measurement-set-name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-B6-A1KN</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>:assay.technology/RNA-seq</td>\n",
       "      <td>RNA-seq data</td>\n",
       "      <td>gx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-B6-A1KN</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>:assay.technology/SNP-array</td>\n",
       "      <td>cnv and variants</td>\n",
       "      <td>snp CNV data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-B6-A1KN</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>:assay.technology/WES</td>\n",
       "      <td>WES</td>\n",
       "      <td>baseline mutations</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-B6-A1KN</td>\n",
       "      <td>tcga-b6-a1kn-10a</td>\n",
       "      <td>:assay.technology/RNA-seq</td>\n",
       "      <td>RNA-seq data</td>\n",
       "      <td>immune-cell-deconvolution</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-B6-A1KN</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>:assay.technology/RNA-seq</td>\n",
       "      <td>RNA-seq data</td>\n",
       "      <td>immune-cell-deconvolution</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TCGA-B6-A1KN</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>:assay.technology/miscellaneous</td>\n",
       "      <td>Tumor purity</td>\n",
       "      <td>tumor purity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>TCGA-B6-A1KN</td>\n",
       "      <td>tcga-b6-a1kn-10a</td>\n",
       "      <td>:assay.technology/SNP-array</td>\n",
       "      <td>cnv and variants</td>\n",
       "      <td>snp CNV data</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 35
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-11-07T20:13:27.181950Z",
     "start_time": "2024-11-07T20:13:26.365437Z"
    }
   },
   "cell_type": "code",
   "source": [
    "samples = pqd.samples(\"tcga-brca\", db_name='tcga-brca')\n",
    "samples"
   ],
   "id": "f229b14d787db808",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "             sample-id sample-tumor-type                sample-specimen  \\\n",
       "0     tcga-a2-a0eu-01a            breast  :sample.specimen/fresh-frozen   \n",
       "1     tcga-a2-a0eu-10a               NaN  :sample.specimen/fresh-frozen   \n",
       "2     tcga-a7-a0d9-01a            breast  :sample.specimen/fresh-frozen   \n",
       "3     tcga-a7-a0d9-10a               NaN  :sample.specimen/fresh-frozen   \n",
       "4     tcga-ao-a0jf-01a            breast  :sample.specimen/fresh-frozen   \n",
       "...                ...               ...                            ...   \n",
       "2291  tcga-b6-a402-10a               NaN  :sample.specimen/fresh-frozen   \n",
       "2292  tcga-e9-a1ne-11a               NaN  :sample.specimen/fresh-frozen   \n",
       "2293  tcga-ol-a5da-10a               NaN  :sample.specimen/fresh-frozen   \n",
       "2294  tcga-bh-a5j0-10a               NaN  :sample.specimen/fresh-frozen   \n",
       "2295  tcga-bh-a0b1-01a            breast  :sample.specimen/fresh-frozen   \n",
       "\n",
       "     sample-metastasis                                     sample-uid  \\\n",
       "0                False  [tcga-brca, 1083184581b3c24ac85291d75a832b9a]   \n",
       "1                False  [tcga-brca, e96e9d5e9e41bba015ba8e4daefda967]   \n",
       "2                False  [tcga-brca, 8c1dbb6ba31a61b3b978a1f93dccb86c]   \n",
       "3                False  [tcga-brca, ba8add19b462211346377f6c7835cb6a]   \n",
       "4                False  [tcga-brca, 0bffbcada51f70f1978d207e22089d60]   \n",
       "...                ...                                            ...   \n",
       "2291             False  [tcga-brca, 9ede5e348369b0d13d4c7989eb565e3c]   \n",
       "2292             False  [tcga-brca, 85358f998e91a477db9bc868554ea46c]   \n",
       "2293             False  [tcga-brca, 7882611517385fe91546199a73b523c7]   \n",
       "2294             False  [tcga-brca, 934e529d67c8a4b6e7840615131ce315]   \n",
       "2295             False  [tcga-brca, 78ff94f90b8de61759b454cf020bd168]   \n",
       "\n",
       "               db-id sample-timepoint-timepoint-id sample-subject-subject-id  \\\n",
       "0     17592187518720            sample procurement              TCGA-A2-A0EU   \n",
       "1     17592187518803            sample procurement              TCGA-A2-A0EU   \n",
       "2     17592187518823            sample procurement              TCGA-A7-A0D9   \n",
       "3     17592187519258            sample procurement              TCGA-A7-A0D9   \n",
       "4     17592187519626            sample procurement              TCGA-AO-A0JF   \n",
       "...              ...                           ...                       ...   \n",
       "2291  17592188110342            sample procurement              TCGA-B6-A402   \n",
       "2292  17592188110466            sample procurement              TCGA-E9-A1NE   \n",
       "2293  17592188111439            sample procurement              TCGA-OL-A5DA   \n",
       "2294  17592188112629            sample procurement              TCGA-BH-A5J0   \n",
       "2295  17592188118826            sample procurement              TCGA-BH-A0B1   \n",
       "\n",
       "     sample-study-day-study-day-id  sample-type-db-id  \\\n",
       "0                               35     17592186045420   \n",
       "1                               35     17592186045421   \n",
       "2                               10     17592186045420   \n",
       "3                               10     17592186045421   \n",
       "4                               57     17592186045420   \n",
       "...                            ...                ...   \n",
       "2291                             0     17592186045421   \n",
       "2292                             0     17592186045421   \n",
       "2293                            33     17592186045421   \n",
       "2294                           104     17592186045421   \n",
       "2295                            44     17592186045420   \n",
       "\n",
       "     sample-gdc-anatomic-site-gdc-anatomic-site-name  \n",
       "0                                             Breast  \n",
       "1                                             Breast  \n",
       "2                                             Breast  \n",
       "3                                             Breast  \n",
       "4                                             Breast  \n",
       "...                                              ...  \n",
       "2291                                          Breast  \n",
       "2292                                          Breast  \n",
       "2293                                          Breast  \n",
       "2294                                          Breast  \n",
       "2295                                          Breast  \n",
       "\n",
       "[2296 rows x 11 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>sample-tumor-type</th>\n",
       "      <th>sample-specimen</th>\n",
       "      <th>sample-metastasis</th>\n",
       "      <th>sample-uid</th>\n",
       "      <th>db-id</th>\n",
       "      <th>sample-timepoint-timepoint-id</th>\n",
       "      <th>sample-subject-subject-id</th>\n",
       "      <th>sample-study-day-study-day-id</th>\n",
       "      <th>sample-type-db-id</th>\n",
       "      <th>sample-gdc-anatomic-site-gdc-anatomic-site-name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>tcga-a2-a0eu-01a</td>\n",
       "      <td>breast</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 1083184581b3c24ac85291d75a832b9a]</td>\n",
       "      <td>17592187518720</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-A2-A0EU</td>\n",
       "      <td>35</td>\n",
       "      <td>17592186045420</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>tcga-a2-a0eu-10a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, e96e9d5e9e41bba015ba8e4daefda967]</td>\n",
       "      <td>17592187518803</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-A2-A0EU</td>\n",
       "      <td>35</td>\n",
       "      <td>17592186045421</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>tcga-a7-a0d9-01a</td>\n",
       "      <td>breast</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 8c1dbb6ba31a61b3b978a1f93dccb86c]</td>\n",
       "      <td>17592187518823</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-A7-A0D9</td>\n",
       "      <td>10</td>\n",
       "      <td>17592186045420</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>tcga-a7-a0d9-10a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, ba8add19b462211346377f6c7835cb6a]</td>\n",
       "      <td>17592187519258</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-A7-A0D9</td>\n",
       "      <td>10</td>\n",
       "      <td>17592186045421</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>tcga-ao-a0jf-01a</td>\n",
       "      <td>breast</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 0bffbcada51f70f1978d207e22089d60]</td>\n",
       "      <td>17592187519626</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-AO-A0JF</td>\n",
       "      <td>57</td>\n",
       "      <td>17592186045420</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2291</th>\n",
       "      <td>tcga-b6-a402-10a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 9ede5e348369b0d13d4c7989eb565e3c]</td>\n",
       "      <td>17592188110342</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-B6-A402</td>\n",
       "      <td>0</td>\n",
       "      <td>17592186045421</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2292</th>\n",
       "      <td>tcga-e9-a1ne-11a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 85358f998e91a477db9bc868554ea46c]</td>\n",
       "      <td>17592188110466</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-E9-A1NE</td>\n",
       "      <td>0</td>\n",
       "      <td>17592186045421</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2293</th>\n",
       "      <td>tcga-ol-a5da-10a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 7882611517385fe91546199a73b523c7]</td>\n",
       "      <td>17592188111439</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-OL-A5DA</td>\n",
       "      <td>33</td>\n",
       "      <td>17592186045421</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2294</th>\n",
       "      <td>tcga-bh-a5j0-10a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 934e529d67c8a4b6e7840615131ce315]</td>\n",
       "      <td>17592188112629</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-BH-A5J0</td>\n",
       "      <td>104</td>\n",
       "      <td>17592186045421</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2295</th>\n",
       "      <td>tcga-bh-a0b1-01a</td>\n",
       "      <td>breast</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 78ff94f90b8de61759b454cf020bd168]</td>\n",
       "      <td>17592188118826</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-BH-A0B1</td>\n",
       "      <td>44</td>\n",
       "      <td>17592186045420</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2296 rows × 11 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 36
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-11-07T20:13:32.075002Z",
     "start_time": "2024-11-07T20:13:32.058070Z"
    }
   },
   "cell_type": "code",
   "source": [
    "samples_sub = samples[samples[\"sample-subject-subject-id\"] == sub_id]\n",
    "samples_sub"
   ],
   "id": "3d4c35905609659e",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "            sample-id sample-tumor-type                sample-specimen  \\\n",
       "105  tcga-b6-a1kn-01a            breast  :sample.specimen/fresh-frozen   \n",
       "106  tcga-b6-a1kn-10a               NaN  :sample.specimen/fresh-frozen   \n",
       "\n",
       "    sample-metastasis                                     sample-uid  \\\n",
       "105             False  [tcga-brca, 83fe2368405c6e3d2a08288d83f8f0db]   \n",
       "106             False  [tcga-brca, 870c4800388cc94b29d58203db3a065b]   \n",
       "\n",
       "              db-id sample-timepoint-timepoint-id sample-subject-subject-id  \\\n",
       "105  17592187547478            sample procurement              TCGA-B6-A1KN   \n",
       "106  17592187548097            sample procurement              TCGA-B6-A1KN   \n",
       "\n",
       "    sample-study-day-study-day-id  sample-type-db-id  \\\n",
       "105                            36     17592186045420   \n",
       "106                            36     17592186045421   \n",
       "\n",
       "    sample-gdc-anatomic-site-gdc-anatomic-site-name  \n",
       "105                                          Breast  \n",
       "106                                          Breast  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>sample-tumor-type</th>\n",
       "      <th>sample-specimen</th>\n",
       "      <th>sample-metastasis</th>\n",
       "      <th>sample-uid</th>\n",
       "      <th>db-id</th>\n",
       "      <th>sample-timepoint-timepoint-id</th>\n",
       "      <th>sample-subject-subject-id</th>\n",
       "      <th>sample-study-day-study-day-id</th>\n",
       "      <th>sample-type-db-id</th>\n",
       "      <th>sample-gdc-anatomic-site-gdc-anatomic-site-name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>breast</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 83fe2368405c6e3d2a08288d83f8f0db]</td>\n",
       "      <td>17592187547478</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-B6-A1KN</td>\n",
       "      <td>36</td>\n",
       "      <td>17592186045420</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>tcga-b6-a1kn-10a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 870c4800388cc94b29d58203db3a065b]</td>\n",
       "      <td>17592187548097</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-B6-A1KN</td>\n",
       "      <td>36</td>\n",
       "      <td>17592186045421</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 37
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-11-07T20:13:35.707858Z",
     "start_time": "2024-11-07T20:13:35.248719Z"
    }
   },
   "cell_type": "code",
   "source": [
    "clin_sum = pqd.clinical_summary(\"tcga-brca\", db_name='tcga-brca')\n",
    "clin_sum"
   ],
   "id": "a2653774454e7b68",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "  clinical-observation-set-name\n",
       "0             pfs os recurrence"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>clinical-observation-set-name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>pfs os recurrence</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 38
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-11-07T20:13:39.168209Z",
     "start_time": "2024-11-07T20:13:38.173793Z"
    }
   },
   "cell_type": "code",
   "source": [
    "sample_for_meas = assays_for_sub[assays_for_sub['measurement-set-name'] == 'baseline mutations']\n",
    "sample_for_meas = sample_for_meas.iloc[0]['sample-id']\n",
    "sample_meas = pqd.sample_measurements('tcga-brca', 'baseline mutations', sample_for_meas, db_name='tcga-brca')\n",
    "sample_meas\n"
   ],
   "id": "44f9e8e55a7f7e88",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "            measurement-variant-variant-id measurement-sample-sample-id\n",
       "0      GRCh37:chr3:+:10500146:10500146/C/T             tcga-b6-a1kn-01a\n",
       "1      GRCh37:chr7:+:25228619:25228619/G/T             tcga-b6-a1kn-01a\n",
       "2     GRCh37:chr20:+:62917053:62917053/G/A             tcga-b6-a1kn-01a\n",
       "3     GRCh37:chr11:+:41032794:41032794/C/T             tcga-b6-a1kn-01a\n",
       "4      GRCh37:chr4:+:24989509:24989509/G/A             tcga-b6-a1kn-01a\n",
       "5   GRCh37:chr20:+:129155089:129155089/C/A             tcga-b6-a1kn-01a\n",
       "6       GRCh37:chr13:+:9783057:9783057/T/G             tcga-b6-a1kn-01a\n",
       "7      GRCh37:chr3:+:68030078:68030078/C/T             tcga-b6-a1kn-01a\n",
       "8      GRCh37:chr8:+:30665245:30665245/G/A             tcga-b6-a1kn-01a\n",
       "9     GRCh37:chr19:+:43853546:43853546/C/T             tcga-b6-a1kn-01a\n",
       "10     GRCh37:chr2:+:16916414:16916414/C/G             tcga-b6-a1kn-01a\n",
       "11  GRCh37:chr14:+:187130348:187130348/C/T             tcga-b6-a1kn-01a\n",
       "12   GRCh37:chr1:+:111739850:111739850/G/A             tcga-b6-a1kn-01a\n",
       "13    GRCh37:chr11:+:38390212:38390212/G/A             tcga-b6-a1kn-01a\n",
       "14     GRCh37:chr3:+:11394113:11394113/C/T             tcga-b6-a1kn-01a\n",
       "15     GRCh37:chr5:+:31714438:31714438/C/A             tcga-b6-a1kn-01a\n",
       "16  GRCh37:chr10:+:176812363:176812363/T/A             tcga-b6-a1kn-01a\n",
       "17     GRCh37:chr8:+:39240627:39240627/T/C             tcga-b6-a1kn-01a\n",
       "18     GRCh37:chr8:+:39261786:39261786/G/A             tcga-b6-a1kn-01a\n",
       "19  GRCh37:chr20:+:153130895:153130895/G/A             tcga-b6-a1kn-01a\n",
       "20     GRCh37:chr4:+:57604136:57604136/C/A             tcga-b6-a1kn-01a\n",
       "21     GRCh37:chr8:+:43592301:43592301/G/A             tcga-b6-a1kn-01a\n",
       "22   GRCh37:chr6:+:100252738:100252738/A/C             tcga-b6-a1kn-01a\n",
       "23     GRCh37:chr6:+:23811029:23811029/G/A             tcga-b6-a1kn-01a\n",
       "24    GRCh37:chr12:+:50512644:50512644/C/T             tcga-b6-a1kn-01a\n",
       "25  GRCh37:chr14:+:146576496:146576496/T/G             tcga-b6-a1kn-01a\n",
       "26  GRCh37:chr10:+:242038987:242038987/A/C             tcga-b6-a1kn-01a\n",
       "27  GRCh37:chr17:+:100679631:100679631/G/A             tcga-b6-a1kn-01a\n",
       "28    GRCh37:chr12:+:38083918:38083918/G/A             tcga-b6-a1kn-01a\n",
       "29    GRCh37:chr10:+:73438003:73438003/C/A             tcga-b6-a1kn-01a\n",
       "30    GRCh37:chr16:+:86181085:86181085/G/T             tcga-b6-a1kn-01a\n",
       "31   GRCh37:chr1:+:205275345:205275345/C/A             tcga-b6-a1kn-01a\n",
       "32       GRCh37:chr9:+:9967532:9967532/C/T             tcga-b6-a1kn-01a\n",
       "33  GRCh37:chr15:+:140559523:140559523/G/A             tcga-b6-a1kn-01a\n",
       "34    GRCh37:chr20:+:18824526:18824526/G/A             tcga-b6-a1kn-01a\n",
       "35  GRCh37:chr17:+:113558745:113558745/C/A             tcga-b6-a1kn-01a\n",
       "36    GRCh37:chr18:+:48697854:48697855/-/T             tcga-b6-a1kn-01a\n",
       "37     GRCh37:chr7:+:31097783:31097783/G/A             tcga-b6-a1kn-01a\n",
       "38   GRCh37:chr1:+:155874570:155874570/A/T             tcga-b6-a1kn-01a\n",
       "39   GRCh37:chr1:+:100573002:100573002/C/A             tcga-b6-a1kn-01a\n",
       "40   GRCh37:chr4:+:100732846:100732846/A/T             tcga-b6-a1kn-01a\n",
       "41     GRCh37:chr4:+:11338943:11338943/G/T             tcga-b6-a1kn-01a\n",
       "42    GRCh37:chr14:+:38029467:38029467/T/C             tcga-b6-a1kn-01a\n",
       "43     GRCh37:chr5:+:75880582:75880582/T/G             tcga-b6-a1kn-01a\n",
       "44  GRCh37:chr15:+:140022616:140022616/A/C             tcga-b6-a1kn-01a\n",
       "45    GRCh37:chr17:+:75617758:75617758/G/A             tcga-b6-a1kn-01a\n",
       "46       GRCh37:chr8:+:7577144:7577144/A/G             tcga-b6-a1kn-01a\n",
       "47     GRCh37:chr5:+:99890758:99890758/T/C             tcga-b6-a1kn-01a\n",
       "48    GRCh37:chr18:+:67578547:67578547/T/G             tcga-b6-a1kn-01a\n",
       "49  GRCh37:chr13:+:101375072:101375072/G/C             tcga-b6-a1kn-01a\n",
       "50    GRCh37:chr15:+:71739680:71739680/G/T             tcga-b6-a1kn-01a\n",
       "51       GRCh37:chr9:+:9770066:9770066/T/C             tcga-b6-a1kn-01a\n",
       "52     GRCh37:chr9:+:38189031:38189031/A/G             tcga-b6-a1kn-01a"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>measurement-variant-variant-id</th>\n",
       "      <th>measurement-sample-sample-id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GRCh37:chr3:+:10500146:10500146/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GRCh37:chr7:+:25228619:25228619/G/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GRCh37:chr20:+:62917053:62917053/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GRCh37:chr11:+:41032794:41032794/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GRCh37:chr4:+:24989509:24989509/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GRCh37:chr20:+:129155089:129155089/C/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GRCh37:chr13:+:9783057:9783057/T/G</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GRCh37:chr3:+:68030078:68030078/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GRCh37:chr8:+:30665245:30665245/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GRCh37:chr19:+:43853546:43853546/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GRCh37:chr2:+:16916414:16916414/C/G</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GRCh37:chr14:+:187130348:187130348/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GRCh37:chr1:+:111739850:111739850/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GRCh37:chr11:+:38390212:38390212/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GRCh37:chr3:+:11394113:11394113/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GRCh37:chr5:+:31714438:31714438/C/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GRCh37:chr10:+:176812363:176812363/T/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GRCh37:chr8:+:39240627:39240627/T/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GRCh37:chr8:+:39261786:39261786/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GRCh37:chr20:+:153130895:153130895/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GRCh37:chr4:+:57604136:57604136/C/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GRCh37:chr8:+:43592301:43592301/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GRCh37:chr6:+:100252738:100252738/A/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GRCh37:chr6:+:23811029:23811029/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GRCh37:chr12:+:50512644:50512644/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GRCh37:chr14:+:146576496:146576496/T/G</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>GRCh37:chr10:+:242038987:242038987/A/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GRCh37:chr17:+:100679631:100679631/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GRCh37:chr12:+:38083918:38083918/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GRCh37:chr10:+:73438003:73438003/C/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>GRCh37:chr16:+:86181085:86181085/G/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GRCh37:chr1:+:205275345:205275345/C/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GRCh37:chr9:+:9967532:9967532/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>GRCh37:chr15:+:140559523:140559523/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>GRCh37:chr20:+:18824526:18824526/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>GRCh37:chr17:+:113558745:113558745/C/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>GRCh37:chr18:+:48697854:48697855/-/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>GRCh37:chr7:+:31097783:31097783/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>GRCh37:chr1:+:155874570:155874570/A/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>GRCh37:chr1:+:100573002:100573002/C/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>GRCh37:chr4:+:100732846:100732846/A/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>GRCh37:chr4:+:11338943:11338943/G/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>GRCh37:chr14:+:38029467:38029467/T/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>GRCh37:chr5:+:75880582:75880582/T/G</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>GRCh37:chr15:+:140022616:140022616/A/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>GRCh37:chr17:+:75617758:75617758/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>GRCh37:chr8:+:7577144:7577144/A/G</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>GRCh37:chr5:+:99890758:99890758/T/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>GRCh37:chr18:+:67578547:67578547/T/G</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>GRCh37:chr13:+:101375072:101375072/G/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>GRCh37:chr15:+:71739680:71739680/G/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>GRCh37:chr9:+:9770066:9770066/T/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>GRCh37:chr9:+:38189031:38189031/A/G</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 39
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-11-07T20:13:41.741188Z",
     "start_time": "2024-11-07T20:13:41.728340Z"
    }
   },
   "cell_type": "code",
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2"
   ],
   "id": "2cf55503ee9d7d18",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    }
   ],
   "execution_count": 40
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-11-07T20:13:50.282462Z",
     "start_time": "2024-11-07T20:13:49.779448Z"
    }
   },
   "cell_type": "code",
   "source": [
    "\n",
    "var = pqr.variants(sample_meas['measurement-variant-variant-id'].tolist(), db_name='tcga-brca')\n",
    "var"
   ],
   "id": "3013e87d1d690659",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "   variant-ref-allele                              variant-id  \\\n",
       "0                   C     GRCh37:chr3:+:10500146:10500146/C/T   \n",
       "1                   G     GRCh37:chr7:+:25228619:25228619/G/T   \n",
       "2                   G    GRCh37:chr20:+:62917053:62917053/G/A   \n",
       "3                   C    GRCh37:chr11:+:41032794:41032794/C/T   \n",
       "4                   G     GRCh37:chr4:+:24989509:24989509/G/A   \n",
       "5                   C  GRCh37:chr20:+:129155089:129155089/C/A   \n",
       "6                   T      GRCh37:chr13:+:9783057:9783057/T/G   \n",
       "7                   C     GRCh37:chr3:+:68030078:68030078/C/T   \n",
       "8                   G     GRCh37:chr8:+:30665245:30665245/G/A   \n",
       "9                   C    GRCh37:chr19:+:43853546:43853546/C/T   \n",
       "10                  C     GRCh37:chr2:+:16916414:16916414/C/G   \n",
       "11                  C  GRCh37:chr14:+:187130348:187130348/C/T   \n",
       "12                  G   GRCh37:chr1:+:111739850:111739850/G/A   \n",
       "13                  G    GRCh37:chr11:+:38390212:38390212/G/A   \n",
       "14                  C     GRCh37:chr3:+:11394113:11394113/C/T   \n",
       "15                  C     GRCh37:chr5:+:31714438:31714438/C/A   \n",
       "16                  T  GRCh37:chr10:+:176812363:176812363/T/A   \n",
       "17                  T     GRCh37:chr8:+:39240627:39240627/T/C   \n",
       "18                  G     GRCh37:chr8:+:39261786:39261786/G/A   \n",
       "19                  G  GRCh37:chr20:+:153130895:153130895/G/A   \n",
       "20                  C     GRCh37:chr4:+:57604136:57604136/C/A   \n",
       "21                  G     GRCh37:chr8:+:43592301:43592301/G/A   \n",
       "22                  A   GRCh37:chr6:+:100252738:100252738/A/C   \n",
       "23                  G     GRCh37:chr6:+:23811029:23811029/G/A   \n",
       "24                  C    GRCh37:chr12:+:50512644:50512644/C/T   \n",
       "25                  T  GRCh37:chr14:+:146576496:146576496/T/G   \n",
       "26                  A  GRCh37:chr10:+:242038987:242038987/A/C   \n",
       "27                  G  GRCh37:chr17:+:100679631:100679631/G/A   \n",
       "28                  G    GRCh37:chr12:+:38083918:38083918/G/A   \n",
       "29                  C    GRCh37:chr10:+:73438003:73438003/C/A   \n",
       "30                  G    GRCh37:chr16:+:86181085:86181085/G/T   \n",
       "31                  C   GRCh37:chr1:+:205275345:205275345/C/A   \n",
       "32                  C       GRCh37:chr9:+:9967532:9967532/C/T   \n",
       "33                  G  GRCh37:chr15:+:140559523:140559523/G/A   \n",
       "34                  G    GRCh37:chr20:+:18824526:18824526/G/A   \n",
       "35                  C  GRCh37:chr17:+:113558745:113558745/C/A   \n",
       "36                  -    GRCh37:chr18:+:48697854:48697855/-/T   \n",
       "37                  G     GRCh37:chr7:+:31097783:31097783/G/A   \n",
       "38                  A   GRCh37:chr1:+:155874570:155874570/A/T   \n",
       "39                  C   GRCh37:chr1:+:100573002:100573002/C/A   \n",
       "40                  A   GRCh37:chr4:+:100732846:100732846/A/T   \n",
       "41                  G     GRCh37:chr4:+:11338943:11338943/G/T   \n",
       "42                  T    GRCh37:chr14:+:38029467:38029467/T/C   \n",
       "43                  T     GRCh37:chr5:+:75880582:75880582/T/G   \n",
       "44                  A  GRCh37:chr15:+:140022616:140022616/A/C   \n",
       "45                  G    GRCh37:chr17:+:75617758:75617758/G/A   \n",
       "46                  A       GRCh37:chr8:+:7577144:7577144/A/G   \n",
       "47                  T     GRCh37:chr5:+:99890758:99890758/T/C   \n",
       "48                  T    GRCh37:chr18:+:67578547:67578547/T/G   \n",
       "49                  G  GRCh37:chr13:+:101375072:101375072/G/C   \n",
       "50                  G    GRCh37:chr15:+:71739680:71739680/G/T   \n",
       "51                  T       GRCh37:chr9:+:9770066:9770066/T/C   \n",
       "52                  A     GRCh37:chr9:+:38189031:38189031/A/G   \n",
       "\n",
       "   variant-alt-allele           db-id               variant-classification  \\\n",
       "0                   T  17592187439906     :variant.classification/missense   \n",
       "1                   T  17592187439908     :variant.classification/missense   \n",
       "2                   A  17592187439910       :variant.classification/silent   \n",
       "3                   T  17592187439912     :variant.classification/missense   \n",
       "4                   A  17592187439914     :variant.classification/missense   \n",
       "5                   A  17592187439917     :variant.classification/missense   \n",
       "6                   G  17592187439919     :variant.classification/missense   \n",
       "7                   T  17592187439921     :variant.classification/missense   \n",
       "8                   A  17592187439923     :variant.classification/missense   \n",
       "9                   T  17592187439925     :variant.classification/missense   \n",
       "10                  G  17592187439927       :variant.classification/silent   \n",
       "11                  T  17592187439929     :variant.classification/missense   \n",
       "12                  A  17592187439931     :variant.classification/missense   \n",
       "13                  A  17592187439933     :variant.classification/missense   \n",
       "14                  T  17592187439935       :variant.classification/silent   \n",
       "15                  A  17592187439937     :variant.classification/missense   \n",
       "16                  A  17592187439939     :variant.classification/missense   \n",
       "17                  C  17592187439941     :variant.classification/missense   \n",
       "18                  A  17592187439943     :variant.classification/missense   \n",
       "19                  A  17592187439945     :variant.classification/missense   \n",
       "20                  A  17592187439947     :variant.classification/missense   \n",
       "21                  A  17592187439949          :variant.classification/RNA   \n",
       "22                  C  17592187439951     :variant.classification/missense   \n",
       "23                  A  17592187439953     :variant.classification/missense   \n",
       "24                  T  17592187439955  :variant.classification/splice-site   \n",
       "25                  G  17592187439957     :variant.classification/missense   \n",
       "26                  C  17592187439959     :variant.classification/missense   \n",
       "27                  A  17592187439961     :variant.classification/missense   \n",
       "28                  A  17592187439963     :variant.classification/missense   \n",
       "29                  A  17592187439965     :variant.classification/missense   \n",
       "30                  T  17592187439967     :variant.classification/missense   \n",
       "31                  A  17592187439969     :variant.classification/missense   \n",
       "32                  T  17592187439971     :variant.classification/missense   \n",
       "33                  A  17592187439973       :variant.classification/silent   \n",
       "34                  A  17592187439975       :variant.classification/silent   \n",
       "35                  A  17592187439977     :variant.classification/nonsense   \n",
       "36                  T  17592187439979                                  NaN   \n",
       "37                  A  17592187439981     :variant.classification/missense   \n",
       "38                  T  17592187439983       :variant.classification/silent   \n",
       "39                  A  17592187439985     :variant.classification/missense   \n",
       "40                  T  17592187439987     :variant.classification/missense   \n",
       "41                  T  17592187439989     :variant.classification/missense   \n",
       "42                  C  17592187439991     :variant.classification/missense   \n",
       "43                  G  17592187439993     :variant.classification/missense   \n",
       "44                  C  17592187439995     :variant.classification/missense   \n",
       "45                  A  17592187439997          :variant.classification/RNA   \n",
       "46                  G  17592187439999     :variant.classification/missense   \n",
       "47                  C  17592187440001     :variant.classification/missense   \n",
       "48                  G  17592187440003     :variant.classification/missense   \n",
       "49                  C  17592187440005     :variant.classification/missense   \n",
       "50                  T  17592187440007     :variant.classification/missense   \n",
       "51                  C  17592187440009     :variant.classification/missense   \n",
       "52                  G  17592187440011       :variant.classification/silent   \n",
       "\n",
       "    variant-type-db-id variant-gene-gene-hgnc-symbol  \\\n",
       "0       17592186045584                        OR5D14   \n",
       "1       17592186045584                       ZCCHC12   \n",
       "2       17592186045584                          FMO3   \n",
       "3       17592186045584                       ATP13A2   \n",
       "4       17592186045584                          CPN2   \n",
       "5       17592186045584                       CNTNAP5   \n",
       "6       17592186045584                         MST1R   \n",
       "7       17592186045584                         CERS1   \n",
       "8       17592186045584                      APOBEC3B   \n",
       "9       17592186045584                         GPAT2   \n",
       "10      17592186045584                      FAM171A1   \n",
       "11      17592186045584                    KRTAP10-10   \n",
       "12      17592186045584                       SLC41A3   \n",
       "13      17592186045584                         ASCC1   \n",
       "14      17592186045584                         RPL41   \n",
       "15      17592186045584                         PEBP4   \n",
       "16      17592186045584                          GYS2   \n",
       "17      17592186045584                          TPH2   \n",
       "18      17592186045584                         CRLF3   \n",
       "19      17592186045584                         SIDT2   \n",
       "20      17592186045584                        PHLPP2   \n",
       "21      17592186045584                         TTBK2   \n",
       "22      17592186045584                           RP9   \n",
       "23      17592186045584                         ASB14   \n",
       "24      17592186045584                         AGAP9   \n",
       "25      17592186045584                        C2CD4C   \n",
       "26      17592186045584                        ACOT11   \n",
       "27      17592186045584                           OGN   \n",
       "28      17592186045584                      SLC25A53   \n",
       "29      17592186045584                          HCN4   \n",
       "30      17592186045584                        ZNF112   \n",
       "31      17592186045584                          UPRT   \n",
       "32      17592186045584                       WFDC10B   \n",
       "33      17592186045584                         GLRA4   \n",
       "34      17592186045584                        NUDT21   \n",
       "35      17592186045584                        CYP2D6   \n",
       "36      17592186045582                      ARHGEF10   \n",
       "37      17592186045584                       CARNMT1   \n",
       "38      17592186045584                         ATXN3   \n",
       "39      17592186045584                      ATP6V1E2   \n",
       "40      17592186045584                         WDTC1   \n",
       "41      17592186045584                       ABHD17C   \n",
       "42      17592186045584                        FBXO46   \n",
       "43      17592186045584                        TANGO6   \n",
       "44      17592186045584                       CBFA2T3   \n",
       "45      17592186045584                       CCDC113   \n",
       "46      17592186045584                        KRBOX4   \n",
       "47      17592186045584                         CCND1   \n",
       "48      17592186045584                         DMBX1   \n",
       "49      17592186045584                          COPE   \n",
       "50      17592186045584                        NFATC4   \n",
       "51      17592186045584                        MMADHC   \n",
       "52      17592186045584                       SIGLEC8   \n",
       "\n",
       "    variant-genomic-coordinates-db-id  unify-import-most-recent-db-id  \n",
       "0                      17592186746250                  17592186177812  \n",
       "1                      17592186746252                  17592186177812  \n",
       "2                      17592186746254                  17592186177812  \n",
       "3                      17592186746256                  17592186177812  \n",
       "4                      17592186746259                  17592186177812  \n",
       "5                      17592186746261                  17592186177812  \n",
       "6                      17592186746263                  17592186177812  \n",
       "7                      17592186746265                  17592186177812  \n",
       "8                      17592186746267                  17592186177812  \n",
       "9                      17592186746269                  17592186177812  \n",
       "10                     17592186746271                  17592186177812  \n",
       "11                     17592186746273                  17592186177812  \n",
       "12                     17592186746275                  17592186177812  \n",
       "13                     17592186746277                  17592186177812  \n",
       "14                     17592186746279                  17592186177812  \n",
       "15                     17592186746281                  17592186177812  \n",
       "16                     17592186746283                  17592186177812  \n",
       "17                     17592186746285                  17592186177812  \n",
       "18                     17592186746287                  17592186177812  \n",
       "19                     17592186746289                  17592186177812  \n",
       "20                     17592186746291                  17592186177812  \n",
       "21                     17592186746293                  17592186177812  \n",
       "22                     17592186746295                  17592186177812  \n",
       "23                     17592186746297                  17592186177812  \n",
       "24                     17592186746299                  17592186177812  \n",
       "25                     17592186746301                  17592186177812  \n",
       "26                     17592186746303                  17592186177812  \n",
       "27                     17592186746305                  17592186177812  \n",
       "28                     17592186746307                  17592186177812  \n",
       "29                     17592186746309                  17592186177812  \n",
       "30                     17592186746311                  17592186177812  \n",
       "31                     17592186746313                  17592186177812  \n",
       "32                     17592186746315                  17592186177812  \n",
       "33                     17592186746317                  17592186177812  \n",
       "34                     17592186746319                  17592186177812  \n",
       "35                     17592186746321                  17592186177812  \n",
       "36                     17592186746323                  17592186177812  \n",
       "37                     17592186746325                  17592186177812  \n",
       "38                     17592186746327                  17592186177812  \n",
       "39                     17592186746329                  17592186177812  \n",
       "40                     17592186746331                  17592186177812  \n",
       "41                     17592186746337                  17592186177812  \n",
       "42                     17592186746339                  17592186177812  \n",
       "43                     17592186746341                  17592186177812  \n",
       "44                     17592186746343                  17592186177812  \n",
       "45                     17592186746345                  17592186177812  \n",
       "46                     17592186728949                  17592186177812  \n",
       "47                     17592186746347                  17592186177812  \n",
       "48                     17592186746349                  17592186177812  \n",
       "49                     17592186746351                  17592186177812  \n",
       "50                     17592186746353                  17592186177812  \n",
       "51                     17592186746355                  17592186177812  \n",
       "52                     17592186746357                  17592186177812  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant-ref-allele</th>\n",
       "      <th>variant-id</th>\n",
       "      <th>variant-alt-allele</th>\n",
       "      <th>db-id</th>\n",
       "      <th>variant-classification</th>\n",
       "      <th>variant-type-db-id</th>\n",
       "      <th>variant-gene-gene-hgnc-symbol</th>\n",
       "      <th>variant-genomic-coordinates-db-id</th>\n",
       "      <th>unify-import-most-recent-db-id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr3:+:10500146:10500146/C/T</td>\n",
       "      <td>T</td>\n",
       "      <td>17592187439906</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>OR5D14</td>\n",
       "      <td>17592186746250</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr7:+:25228619:25228619/G/T</td>\n",
       "      <td>T</td>\n",
       "      <td>17592187439908</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>ZCCHC12</td>\n",
       "      <td>17592186746252</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr20:+:62917053:62917053/G/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439910</td>\n",
       "      <td>:variant.classification/silent</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>FMO3</td>\n",
       "      <td>17592186746254</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr11:+:41032794:41032794/C/T</td>\n",
       "      <td>T</td>\n",
       "      <td>17592187439912</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>ATP13A2</td>\n",
       "      <td>17592186746256</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr4:+:24989509:24989509/G/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439914</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>CPN2</td>\n",
       "      <td>17592186746259</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr20:+:129155089:129155089/C/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439917</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>CNTNAP5</td>\n",
       "      <td>17592186746261</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>T</td>\n",
       "      <td>GRCh37:chr13:+:9783057:9783057/T/G</td>\n",
       "      <td>G</td>\n",
       "      <td>17592187439919</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>MST1R</td>\n",
       "      <td>17592186746263</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr3:+:68030078:68030078/C/T</td>\n",
       "      <td>T</td>\n",
       "      <td>17592187439921</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>CERS1</td>\n",
       "      <td>17592186746265</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr8:+:30665245:30665245/G/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439923</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>APOBEC3B</td>\n",
       "      <td>17592186746267</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr19:+:43853546:43853546/C/T</td>\n",
       "      <td>T</td>\n",
       "      <td>17592187439925</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>GPAT2</td>\n",
       "      <td>17592186746269</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr2:+:16916414:16916414/C/G</td>\n",
       "      <td>G</td>\n",
       "      <td>17592187439927</td>\n",
       "      <td>:variant.classification/silent</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>FAM171A1</td>\n",
       "      <td>17592186746271</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr14:+:187130348:187130348/C/T</td>\n",
       "      <td>T</td>\n",
       "      <td>17592187439929</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>KRTAP10-10</td>\n",
       "      <td>17592186746273</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr1:+:111739850:111739850/G/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439931</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>SLC41A3</td>\n",
       "      <td>17592186746275</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr11:+:38390212:38390212/G/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439933</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>ASCC1</td>\n",
       "      <td>17592186746277</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr3:+:11394113:11394113/C/T</td>\n",
       "      <td>T</td>\n",
       "      <td>17592187439935</td>\n",
       "      <td>:variant.classification/silent</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>RPL41</td>\n",
       "      <td>17592186746279</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr5:+:31714438:31714438/C/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439937</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>PEBP4</td>\n",
       "      <td>17592186746281</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>T</td>\n",
       "      <td>GRCh37:chr10:+:176812363:176812363/T/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439939</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>GYS2</td>\n",
       "      <td>17592186746283</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>T</td>\n",
       "      <td>GRCh37:chr8:+:39240627:39240627/T/C</td>\n",
       "      <td>C</td>\n",
       "      <td>17592187439941</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>TPH2</td>\n",
       "      <td>17592186746285</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr8:+:39261786:39261786/G/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439943</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>CRLF3</td>\n",
       "      <td>17592186746287</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr20:+:153130895:153130895/G/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439945</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>SIDT2</td>\n",
       "      <td>17592186746289</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr4:+:57604136:57604136/C/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439947</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>PHLPP2</td>\n",
       "      <td>17592186746291</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr8:+:43592301:43592301/G/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439949</td>\n",
       "      <td>:variant.classification/RNA</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>TTBK2</td>\n",
       "      <td>17592186746293</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>A</td>\n",
       "      <td>GRCh37:chr6:+:100252738:100252738/A/C</td>\n",
       "      <td>C</td>\n",
       "      <td>17592187439951</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>RP9</td>\n",
       "      <td>17592186746295</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr6:+:23811029:23811029/G/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439953</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>ASB14</td>\n",
       "      <td>17592186746297</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr12:+:50512644:50512644/C/T</td>\n",
       "      <td>T</td>\n",
       "      <td>17592187439955</td>\n",
       "      <td>:variant.classification/splice-site</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>AGAP9</td>\n",
       "      <td>17592186746299</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>T</td>\n",
       "      <td>GRCh37:chr14:+:146576496:146576496/T/G</td>\n",
       "      <td>G</td>\n",
       "      <td>17592187439957</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>C2CD4C</td>\n",
       "      <td>17592186746301</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>A</td>\n",
       "      <td>GRCh37:chr10:+:242038987:242038987/A/C</td>\n",
       "      <td>C</td>\n",
       "      <td>17592187439959</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>ACOT11</td>\n",
       "      <td>17592186746303</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr17:+:100679631:100679631/G/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439961</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>OGN</td>\n",
       "      <td>17592186746305</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr12:+:38083918:38083918/G/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439963</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>SLC25A53</td>\n",
       "      <td>17592186746307</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr10:+:73438003:73438003/C/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439965</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>HCN4</td>\n",
       "      <td>17592186746309</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr16:+:86181085:86181085/G/T</td>\n",
       "      <td>T</td>\n",
       "      <td>17592187439967</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>ZNF112</td>\n",
       "      <td>17592186746311</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr1:+:205275345:205275345/C/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439969</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>UPRT</td>\n",
       "      <td>17592186746313</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr9:+:9967532:9967532/C/T</td>\n",
       "      <td>T</td>\n",
       "      <td>17592187439971</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>WFDC10B</td>\n",
       "      <td>17592186746315</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr15:+:140559523:140559523/G/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439973</td>\n",
       "      <td>:variant.classification/silent</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>GLRA4</td>\n",
       "      <td>17592186746317</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr20:+:18824526:18824526/G/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439975</td>\n",
       "      <td>:variant.classification/silent</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>NUDT21</td>\n",
       "      <td>17592186746319</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr17:+:113558745:113558745/C/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439977</td>\n",
       "      <td>:variant.classification/nonsense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>CYP2D6</td>\n",
       "      <td>17592186746321</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>-</td>\n",
       "      <td>GRCh37:chr18:+:48697854:48697855/-/T</td>\n",
       "      <td>T</td>\n",
       "      <td>17592187439979</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17592186045582</td>\n",
       "      <td>ARHGEF10</td>\n",
       "      <td>17592186746323</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr7:+:31097783:31097783/G/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439981</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>CARNMT1</td>\n",
       "      <td>17592186746325</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>A</td>\n",
       "      <td>GRCh37:chr1:+:155874570:155874570/A/T</td>\n",
       "      <td>T</td>\n",
       "      <td>17592187439983</td>\n",
       "      <td>:variant.classification/silent</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>ATXN3</td>\n",
       "      <td>17592186746327</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr1:+:100573002:100573002/C/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439985</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>ATP6V1E2</td>\n",
       "      <td>17592186746329</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>A</td>\n",
       "      <td>GRCh37:chr4:+:100732846:100732846/A/T</td>\n",
       "      <td>T</td>\n",
       "      <td>17592187439987</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>WDTC1</td>\n",
       "      <td>17592186746331</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr4:+:11338943:11338943/G/T</td>\n",
       "      <td>T</td>\n",
       "      <td>17592187439989</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>ABHD17C</td>\n",
       "      <td>17592186746337</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>T</td>\n",
       "      <td>GRCh37:chr14:+:38029467:38029467/T/C</td>\n",
       "      <td>C</td>\n",
       "      <td>17592187439991</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>FBXO46</td>\n",
       "      <td>17592186746339</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>T</td>\n",
       "      <td>GRCh37:chr5:+:75880582:75880582/T/G</td>\n",
       "      <td>G</td>\n",
       "      <td>17592187439993</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>TANGO6</td>\n",
       "      <td>17592186746341</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>A</td>\n",
       "      <td>GRCh37:chr15:+:140022616:140022616/A/C</td>\n",
       "      <td>C</td>\n",
       "      <td>17592187439995</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>CBFA2T3</td>\n",
       "      <td>17592186746343</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr17:+:75617758:75617758/G/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187439997</td>\n",
       "      <td>:variant.classification/RNA</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>CCDC113</td>\n",
       "      <td>17592186746345</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>A</td>\n",
       "      <td>GRCh37:chr8:+:7577144:7577144/A/G</td>\n",
       "      <td>G</td>\n",
       "      <td>17592187439999</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>KRBOX4</td>\n",
       "      <td>17592186728949</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>T</td>\n",
       "      <td>GRCh37:chr5:+:99890758:99890758/T/C</td>\n",
       "      <td>C</td>\n",
       "      <td>17592187440001</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>CCND1</td>\n",
       "      <td>17592186746347</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>T</td>\n",
       "      <td>GRCh37:chr18:+:67578547:67578547/T/G</td>\n",
       "      <td>G</td>\n",
       "      <td>17592187440003</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>DMBX1</td>\n",
       "      <td>17592186746349</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr13:+:101375072:101375072/G/C</td>\n",
       "      <td>C</td>\n",
       "      <td>17592187440005</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>COPE</td>\n",
       "      <td>17592186746351</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr15:+:71739680:71739680/G/T</td>\n",
       "      <td>T</td>\n",
       "      <td>17592187440007</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>NFATC4</td>\n",
       "      <td>17592186746353</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>T</td>\n",
       "      <td>GRCh37:chr9:+:9770066:9770066/T/C</td>\n",
       "      <td>C</td>\n",
       "      <td>17592187440009</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>MMADHC</td>\n",
       "      <td>17592186746355</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>A</td>\n",
       "      <td>GRCh37:chr9:+:38189031:38189031/A/G</td>\n",
       "      <td>G</td>\n",
       "      <td>17592187440011</td>\n",
       "      <td>:variant.classification/silent</td>\n",
       "      <td>17592186045584</td>\n",
       "      <td>SIGLEC8</td>\n",
       "      <td>17592186746357</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 42
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-11-07T20:15:03.412991Z",
     "start_time": "2024-11-07T20:15:03.381327Z"
    }
   },
   "cell_type": "code",
   "source": [
    "w_var = sample_meas.merge(\n",
    "    var[['variant-id',\n",
    "         'variant-ref-allele',\n",
    "         'variant-alt-allele',\n",
    "         'variant-gene-gene-hgnc-symbol',\n",
    "         'variant-classification']],\n",
    "    left_on='measurement-variant-variant-id',\n",
    "    right_on='variant-id',\n",
    "    how='left'\n",
    ")\n",
    "w_var"
   ],
   "id": "9c34888ffd0dd9ca",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "            measurement-variant-variant-id measurement-sample-sample-id  \\\n",
       "0      GRCh37:chr3:+:10500146:10500146/C/T             tcga-b6-a1kn-01a   \n",
       "1      GRCh37:chr7:+:25228619:25228619/G/T             tcga-b6-a1kn-01a   \n",
       "2     GRCh37:chr20:+:62917053:62917053/G/A             tcga-b6-a1kn-01a   \n",
       "3     GRCh37:chr11:+:41032794:41032794/C/T             tcga-b6-a1kn-01a   \n",
       "4      GRCh37:chr4:+:24989509:24989509/G/A             tcga-b6-a1kn-01a   \n",
       "5   GRCh37:chr20:+:129155089:129155089/C/A             tcga-b6-a1kn-01a   \n",
       "6       GRCh37:chr13:+:9783057:9783057/T/G             tcga-b6-a1kn-01a   \n",
       "7      GRCh37:chr3:+:68030078:68030078/C/T             tcga-b6-a1kn-01a   \n",
       "8      GRCh37:chr8:+:30665245:30665245/G/A             tcga-b6-a1kn-01a   \n",
       "9     GRCh37:chr19:+:43853546:43853546/C/T             tcga-b6-a1kn-01a   \n",
       "10     GRCh37:chr2:+:16916414:16916414/C/G             tcga-b6-a1kn-01a   \n",
       "11  GRCh37:chr14:+:187130348:187130348/C/T             tcga-b6-a1kn-01a   \n",
       "12   GRCh37:chr1:+:111739850:111739850/G/A             tcga-b6-a1kn-01a   \n",
       "13    GRCh37:chr11:+:38390212:38390212/G/A             tcga-b6-a1kn-01a   \n",
       "14     GRCh37:chr3:+:11394113:11394113/C/T             tcga-b6-a1kn-01a   \n",
       "15     GRCh37:chr5:+:31714438:31714438/C/A             tcga-b6-a1kn-01a   \n",
       "16  GRCh37:chr10:+:176812363:176812363/T/A             tcga-b6-a1kn-01a   \n",
       "17     GRCh37:chr8:+:39240627:39240627/T/C             tcga-b6-a1kn-01a   \n",
       "18     GRCh37:chr8:+:39261786:39261786/G/A             tcga-b6-a1kn-01a   \n",
       "19  GRCh37:chr20:+:153130895:153130895/G/A             tcga-b6-a1kn-01a   \n",
       "20     GRCh37:chr4:+:57604136:57604136/C/A             tcga-b6-a1kn-01a   \n",
       "21     GRCh37:chr8:+:43592301:43592301/G/A             tcga-b6-a1kn-01a   \n",
       "22   GRCh37:chr6:+:100252738:100252738/A/C             tcga-b6-a1kn-01a   \n",
       "23     GRCh37:chr6:+:23811029:23811029/G/A             tcga-b6-a1kn-01a   \n",
       "24    GRCh37:chr12:+:50512644:50512644/C/T             tcga-b6-a1kn-01a   \n",
       "25  GRCh37:chr14:+:146576496:146576496/T/G             tcga-b6-a1kn-01a   \n",
       "26  GRCh37:chr10:+:242038987:242038987/A/C             tcga-b6-a1kn-01a   \n",
       "27  GRCh37:chr17:+:100679631:100679631/G/A             tcga-b6-a1kn-01a   \n",
       "28    GRCh37:chr12:+:38083918:38083918/G/A             tcga-b6-a1kn-01a   \n",
       "29    GRCh37:chr10:+:73438003:73438003/C/A             tcga-b6-a1kn-01a   \n",
       "30    GRCh37:chr16:+:86181085:86181085/G/T             tcga-b6-a1kn-01a   \n",
       "31   GRCh37:chr1:+:205275345:205275345/C/A             tcga-b6-a1kn-01a   \n",
       "32       GRCh37:chr9:+:9967532:9967532/C/T             tcga-b6-a1kn-01a   \n",
       "33  GRCh37:chr15:+:140559523:140559523/G/A             tcga-b6-a1kn-01a   \n",
       "34    GRCh37:chr20:+:18824526:18824526/G/A             tcga-b6-a1kn-01a   \n",
       "35  GRCh37:chr17:+:113558745:113558745/C/A             tcga-b6-a1kn-01a   \n",
       "36    GRCh37:chr18:+:48697854:48697855/-/T             tcga-b6-a1kn-01a   \n",
       "37     GRCh37:chr7:+:31097783:31097783/G/A             tcga-b6-a1kn-01a   \n",
       "38   GRCh37:chr1:+:155874570:155874570/A/T             tcga-b6-a1kn-01a   \n",
       "39   GRCh37:chr1:+:100573002:100573002/C/A             tcga-b6-a1kn-01a   \n",
       "40   GRCh37:chr4:+:100732846:100732846/A/T             tcga-b6-a1kn-01a   \n",
       "41     GRCh37:chr4:+:11338943:11338943/G/T             tcga-b6-a1kn-01a   \n",
       "42    GRCh37:chr14:+:38029467:38029467/T/C             tcga-b6-a1kn-01a   \n",
       "43     GRCh37:chr5:+:75880582:75880582/T/G             tcga-b6-a1kn-01a   \n",
       "44  GRCh37:chr15:+:140022616:140022616/A/C             tcga-b6-a1kn-01a   \n",
       "45    GRCh37:chr17:+:75617758:75617758/G/A             tcga-b6-a1kn-01a   \n",
       "46       GRCh37:chr8:+:7577144:7577144/A/G             tcga-b6-a1kn-01a   \n",
       "47     GRCh37:chr5:+:99890758:99890758/T/C             tcga-b6-a1kn-01a   \n",
       "48    GRCh37:chr18:+:67578547:67578547/T/G             tcga-b6-a1kn-01a   \n",
       "49  GRCh37:chr13:+:101375072:101375072/G/C             tcga-b6-a1kn-01a   \n",
       "50    GRCh37:chr15:+:71739680:71739680/G/T             tcga-b6-a1kn-01a   \n",
       "51       GRCh37:chr9:+:9770066:9770066/T/C             tcga-b6-a1kn-01a   \n",
       "52     GRCh37:chr9:+:38189031:38189031/A/G             tcga-b6-a1kn-01a   \n",
       "\n",
       "                                variant-id variant-ref-allele  \\\n",
       "0      GRCh37:chr3:+:10500146:10500146/C/T                  C   \n",
       "1      GRCh37:chr7:+:25228619:25228619/G/T                  G   \n",
       "2     GRCh37:chr20:+:62917053:62917053/G/A                  G   \n",
       "3     GRCh37:chr11:+:41032794:41032794/C/T                  C   \n",
       "4      GRCh37:chr4:+:24989509:24989509/G/A                  G   \n",
       "5   GRCh37:chr20:+:129155089:129155089/C/A                  C   \n",
       "6       GRCh37:chr13:+:9783057:9783057/T/G                  T   \n",
       "7      GRCh37:chr3:+:68030078:68030078/C/T                  C   \n",
       "8      GRCh37:chr8:+:30665245:30665245/G/A                  G   \n",
       "9     GRCh37:chr19:+:43853546:43853546/C/T                  C   \n",
       "10     GRCh37:chr2:+:16916414:16916414/C/G                  C   \n",
       "11  GRCh37:chr14:+:187130348:187130348/C/T                  C   \n",
       "12   GRCh37:chr1:+:111739850:111739850/G/A                  G   \n",
       "13    GRCh37:chr11:+:38390212:38390212/G/A                  G   \n",
       "14     GRCh37:chr3:+:11394113:11394113/C/T                  C   \n",
       "15     GRCh37:chr5:+:31714438:31714438/C/A                  C   \n",
       "16  GRCh37:chr10:+:176812363:176812363/T/A                  T   \n",
       "17     GRCh37:chr8:+:39240627:39240627/T/C                  T   \n",
       "18     GRCh37:chr8:+:39261786:39261786/G/A                  G   \n",
       "19  GRCh37:chr20:+:153130895:153130895/G/A                  G   \n",
       "20     GRCh37:chr4:+:57604136:57604136/C/A                  C   \n",
       "21     GRCh37:chr8:+:43592301:43592301/G/A                  G   \n",
       "22   GRCh37:chr6:+:100252738:100252738/A/C                  A   \n",
       "23     GRCh37:chr6:+:23811029:23811029/G/A                  G   \n",
       "24    GRCh37:chr12:+:50512644:50512644/C/T                  C   \n",
       "25  GRCh37:chr14:+:146576496:146576496/T/G                  T   \n",
       "26  GRCh37:chr10:+:242038987:242038987/A/C                  A   \n",
       "27  GRCh37:chr17:+:100679631:100679631/G/A                  G   \n",
       "28    GRCh37:chr12:+:38083918:38083918/G/A                  G   \n",
       "29    GRCh37:chr10:+:73438003:73438003/C/A                  C   \n",
       "30    GRCh37:chr16:+:86181085:86181085/G/T                  G   \n",
       "31   GRCh37:chr1:+:205275345:205275345/C/A                  C   \n",
       "32       GRCh37:chr9:+:9967532:9967532/C/T                  C   \n",
       "33  GRCh37:chr15:+:140559523:140559523/G/A                  G   \n",
       "34    GRCh37:chr20:+:18824526:18824526/G/A                  G   \n",
       "35  GRCh37:chr17:+:113558745:113558745/C/A                  C   \n",
       "36    GRCh37:chr18:+:48697854:48697855/-/T                  -   \n",
       "37     GRCh37:chr7:+:31097783:31097783/G/A                  G   \n",
       "38   GRCh37:chr1:+:155874570:155874570/A/T                  A   \n",
       "39   GRCh37:chr1:+:100573002:100573002/C/A                  C   \n",
       "40   GRCh37:chr4:+:100732846:100732846/A/T                  A   \n",
       "41     GRCh37:chr4:+:11338943:11338943/G/T                  G   \n",
       "42    GRCh37:chr14:+:38029467:38029467/T/C                  T   \n",
       "43     GRCh37:chr5:+:75880582:75880582/T/G                  T   \n",
       "44  GRCh37:chr15:+:140022616:140022616/A/C                  A   \n",
       "45    GRCh37:chr17:+:75617758:75617758/G/A                  G   \n",
       "46       GRCh37:chr8:+:7577144:7577144/A/G                  A   \n",
       "47     GRCh37:chr5:+:99890758:99890758/T/C                  T   \n",
       "48    GRCh37:chr18:+:67578547:67578547/T/G                  T   \n",
       "49  GRCh37:chr13:+:101375072:101375072/G/C                  G   \n",
       "50    GRCh37:chr15:+:71739680:71739680/G/T                  G   \n",
       "51       GRCh37:chr9:+:9770066:9770066/T/C                  T   \n",
       "52     GRCh37:chr9:+:38189031:38189031/A/G                  A   \n",
       "\n",
       "   variant-alt-allele variant-gene-gene-hgnc-symbol  \\\n",
       "0                   T                        OR5D14   \n",
       "1                   T                       ZCCHC12   \n",
       "2                   A                          FMO3   \n",
       "3                   T                       ATP13A2   \n",
       "4                   A                          CPN2   \n",
       "5                   A                       CNTNAP5   \n",
       "6                   G                         MST1R   \n",
       "7                   T                         CERS1   \n",
       "8                   A                      APOBEC3B   \n",
       "9                   T                         GPAT2   \n",
       "10                  G                      FAM171A1   \n",
       "11                  T                    KRTAP10-10   \n",
       "12                  A                       SLC41A3   \n",
       "13                  A                         ASCC1   \n",
       "14                  T                         RPL41   \n",
       "15                  A                         PEBP4   \n",
       "16                  A                          GYS2   \n",
       "17                  C                          TPH2   \n",
       "18                  A                         CRLF3   \n",
       "19                  A                         SIDT2   \n",
       "20                  A                        PHLPP2   \n",
       "21                  A                         TTBK2   \n",
       "22                  C                           RP9   \n",
       "23                  A                         ASB14   \n",
       "24                  T                         AGAP9   \n",
       "25                  G                        C2CD4C   \n",
       "26                  C                        ACOT11   \n",
       "27                  A                           OGN   \n",
       "28                  A                      SLC25A53   \n",
       "29                  A                          HCN4   \n",
       "30                  T                        ZNF112   \n",
       "31                  A                          UPRT   \n",
       "32                  T                       WFDC10B   \n",
       "33                  A                         GLRA4   \n",
       "34                  A                        NUDT21   \n",
       "35                  A                        CYP2D6   \n",
       "36                  T                      ARHGEF10   \n",
       "37                  A                       CARNMT1   \n",
       "38                  T                         ATXN3   \n",
       "39                  A                      ATP6V1E2   \n",
       "40                  T                         WDTC1   \n",
       "41                  T                       ABHD17C   \n",
       "42                  C                        FBXO46   \n",
       "43                  G                        TANGO6   \n",
       "44                  C                       CBFA2T3   \n",
       "45                  A                       CCDC113   \n",
       "46                  G                        KRBOX4   \n",
       "47                  C                         CCND1   \n",
       "48                  G                         DMBX1   \n",
       "49                  C                          COPE   \n",
       "50                  T                        NFATC4   \n",
       "51                  C                        MMADHC   \n",
       "52                  G                       SIGLEC8   \n",
       "\n",
       "                 variant-classification  \n",
       "0      :variant.classification/missense  \n",
       "1      :variant.classification/missense  \n",
       "2        :variant.classification/silent  \n",
       "3      :variant.classification/missense  \n",
       "4      :variant.classification/missense  \n",
       "5      :variant.classification/missense  \n",
       "6      :variant.classification/missense  \n",
       "7      :variant.classification/missense  \n",
       "8      :variant.classification/missense  \n",
       "9      :variant.classification/missense  \n",
       "10       :variant.classification/silent  \n",
       "11     :variant.classification/missense  \n",
       "12     :variant.classification/missense  \n",
       "13     :variant.classification/missense  \n",
       "14       :variant.classification/silent  \n",
       "15     :variant.classification/missense  \n",
       "16     :variant.classification/missense  \n",
       "17     :variant.classification/missense  \n",
       "18     :variant.classification/missense  \n",
       "19     :variant.classification/missense  \n",
       "20     :variant.classification/missense  \n",
       "21          :variant.classification/RNA  \n",
       "22     :variant.classification/missense  \n",
       "23     :variant.classification/missense  \n",
       "24  :variant.classification/splice-site  \n",
       "25     :variant.classification/missense  \n",
       "26     :variant.classification/missense  \n",
       "27     :variant.classification/missense  \n",
       "28     :variant.classification/missense  \n",
       "29     :variant.classification/missense  \n",
       "30     :variant.classification/missense  \n",
       "31     :variant.classification/missense  \n",
       "32     :variant.classification/missense  \n",
       "33       :variant.classification/silent  \n",
       "34       :variant.classification/silent  \n",
       "35     :variant.classification/nonsense  \n",
       "36                                  NaN  \n",
       "37     :variant.classification/missense  \n",
       "38       :variant.classification/silent  \n",
       "39     :variant.classification/missense  \n",
       "40     :variant.classification/missense  \n",
       "41     :variant.classification/missense  \n",
       "42     :variant.classification/missense  \n",
       "43     :variant.classification/missense  \n",
       "44     :variant.classification/missense  \n",
       "45          :variant.classification/RNA  \n",
       "46     :variant.classification/missense  \n",
       "47     :variant.classification/missense  \n",
       "48     :variant.classification/missense  \n",
       "49     :variant.classification/missense  \n",
       "50     :variant.classification/missense  \n",
       "51     :variant.classification/missense  \n",
       "52       :variant.classification/silent  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>measurement-variant-variant-id</th>\n",
       "      <th>measurement-sample-sample-id</th>\n",
       "      <th>variant-id</th>\n",
       "      <th>variant-ref-allele</th>\n",
       "      <th>variant-alt-allele</th>\n",
       "      <th>variant-gene-gene-hgnc-symbol</th>\n",
       "      <th>variant-classification</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GRCh37:chr3:+:10500146:10500146/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr3:+:10500146:10500146/C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>OR5D14</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GRCh37:chr7:+:25228619:25228619/G/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr7:+:25228619:25228619/G/T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ZCCHC12</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GRCh37:chr20:+:62917053:62917053/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr20:+:62917053:62917053/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>FMO3</td>\n",
       "      <td>:variant.classification/silent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GRCh37:chr11:+:41032794:41032794/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr11:+:41032794:41032794/C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ATP13A2</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GRCh37:chr4:+:24989509:24989509/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr4:+:24989509:24989509/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CPN2</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GRCh37:chr20:+:129155089:129155089/C/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr20:+:129155089:129155089/C/A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>CNTNAP5</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GRCh37:chr13:+:9783057:9783057/T/G</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr13:+:9783057:9783057/T/G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>MST1R</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GRCh37:chr3:+:68030078:68030078/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr3:+:68030078:68030078/C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CERS1</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GRCh37:chr8:+:30665245:30665245/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr8:+:30665245:30665245/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>APOBEC3B</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GRCh37:chr19:+:43853546:43853546/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr19:+:43853546:43853546/C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>GPAT2</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GRCh37:chr2:+:16916414:16916414/C/G</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr2:+:16916414:16916414/C/G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>FAM171A1</td>\n",
       "      <td>:variant.classification/silent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GRCh37:chr14:+:187130348:187130348/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr14:+:187130348:187130348/C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>KRTAP10-10</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GRCh37:chr1:+:111739850:111739850/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr1:+:111739850:111739850/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SLC41A3</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GRCh37:chr11:+:38390212:38390212/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr11:+:38390212:38390212/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ASCC1</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GRCh37:chr3:+:11394113:11394113/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr3:+:11394113:11394113/C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>RPL41</td>\n",
       "      <td>:variant.classification/silent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GRCh37:chr5:+:31714438:31714438/C/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr5:+:31714438:31714438/C/A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>PEBP4</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GRCh37:chr10:+:176812363:176812363/T/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr10:+:176812363:176812363/T/A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>GYS2</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GRCh37:chr8:+:39240627:39240627/T/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr8:+:39240627:39240627/T/C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>TPH2</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GRCh37:chr8:+:39261786:39261786/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr8:+:39261786:39261786/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CRLF3</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GRCh37:chr20:+:153130895:153130895/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr20:+:153130895:153130895/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SIDT2</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GRCh37:chr4:+:57604136:57604136/C/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr4:+:57604136:57604136/C/A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>PHLPP2</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GRCh37:chr8:+:43592301:43592301/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr8:+:43592301:43592301/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>TTBK2</td>\n",
       "      <td>:variant.classification/RNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GRCh37:chr6:+:100252738:100252738/A/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr6:+:100252738:100252738/A/C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>RP9</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GRCh37:chr6:+:23811029:23811029/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr6:+:23811029:23811029/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ASB14</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GRCh37:chr12:+:50512644:50512644/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr12:+:50512644:50512644/C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>AGAP9</td>\n",
       "      <td>:variant.classification/splice-site</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GRCh37:chr14:+:146576496:146576496/T/G</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr14:+:146576496:146576496/T/G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>C2CD4C</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>GRCh37:chr10:+:242038987:242038987/A/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr10:+:242038987:242038987/A/C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>ACOT11</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GRCh37:chr17:+:100679631:100679631/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr17:+:100679631:100679631/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>OGN</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GRCh37:chr12:+:38083918:38083918/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr12:+:38083918:38083918/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SLC25A53</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GRCh37:chr10:+:73438003:73438003/C/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr10:+:73438003:73438003/C/A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>HCN4</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>GRCh37:chr16:+:86181085:86181085/G/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr16:+:86181085:86181085/G/T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ZNF112</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GRCh37:chr1:+:205275345:205275345/C/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr1:+:205275345:205275345/C/A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>UPRT</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GRCh37:chr9:+:9967532:9967532/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr9:+:9967532:9967532/C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>WFDC10B</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>GRCh37:chr15:+:140559523:140559523/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr15:+:140559523:140559523/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>GLRA4</td>\n",
       "      <td>:variant.classification/silent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>GRCh37:chr20:+:18824526:18824526/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr20:+:18824526:18824526/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NUDT21</td>\n",
       "      <td>:variant.classification/silent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>GRCh37:chr17:+:113558745:113558745/C/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr17:+:113558745:113558745/C/A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>CYP2D6</td>\n",
       "      <td>:variant.classification/nonsense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>GRCh37:chr18:+:48697854:48697855/-/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr18:+:48697854:48697855/-/T</td>\n",
       "      <td>-</td>\n",
       "      <td>T</td>\n",
       "      <td>ARHGEF10</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>GRCh37:chr7:+:31097783:31097783/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr7:+:31097783:31097783/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CARNMT1</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>GRCh37:chr1:+:155874570:155874570/A/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr1:+:155874570:155874570/A/T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>ATXN3</td>\n",
       "      <td>:variant.classification/silent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>GRCh37:chr1:+:100573002:100573002/C/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr1:+:100573002:100573002/C/A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>ATP6V1E2</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>GRCh37:chr4:+:100732846:100732846/A/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr4:+:100732846:100732846/A/T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>WDTC1</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>GRCh37:chr4:+:11338943:11338943/G/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr4:+:11338943:11338943/G/T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ABHD17C</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>GRCh37:chr14:+:38029467:38029467/T/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr14:+:38029467:38029467/T/C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>FBXO46</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>GRCh37:chr5:+:75880582:75880582/T/G</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr5:+:75880582:75880582/T/G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>TANGO6</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>GRCh37:chr15:+:140022616:140022616/A/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr15:+:140022616:140022616/A/C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>CBFA2T3</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>GRCh37:chr17:+:75617758:75617758/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr17:+:75617758:75617758/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CCDC113</td>\n",
       "      <td>:variant.classification/RNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>GRCh37:chr8:+:7577144:7577144/A/G</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr8:+:7577144:7577144/A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>KRBOX4</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>GRCh37:chr5:+:99890758:99890758/T/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr5:+:99890758:99890758/T/C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>CCND1</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>GRCh37:chr18:+:67578547:67578547/T/G</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr18:+:67578547:67578547/T/G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>DMBX1</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>GRCh37:chr13:+:101375072:101375072/G/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr13:+:101375072:101375072/G/C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>COPE</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>GRCh37:chr15:+:71739680:71739680/G/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr15:+:71739680:71739680/G/T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NFATC4</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>GRCh37:chr9:+:9770066:9770066/T/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr9:+:9770066:9770066/T/C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>MMADHC</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>GRCh37:chr9:+:38189031:38189031/A/G</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr9:+:38189031:38189031/A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SIGLEC8</td>\n",
       "      <td>:variant.classification/silent</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 43
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-11-07T20:16:18.716210Z",
     "start_time": "2024-11-07T20:16:18.702202Z"
    }
   },
   "cell_type": "code",
   "source": "w_var[w_var['variant-classification'] == ':variant.classification/missense']",
   "id": "8f53087db8f6e263",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "            measurement-variant-variant-id measurement-sample-sample-id  \\\n",
       "0      GRCh37:chr3:+:10500146:10500146/C/T             tcga-b6-a1kn-01a   \n",
       "1      GRCh37:chr7:+:25228619:25228619/G/T             tcga-b6-a1kn-01a   \n",
       "3     GRCh37:chr11:+:41032794:41032794/C/T             tcga-b6-a1kn-01a   \n",
       "4      GRCh37:chr4:+:24989509:24989509/G/A             tcga-b6-a1kn-01a   \n",
       "5   GRCh37:chr20:+:129155089:129155089/C/A             tcga-b6-a1kn-01a   \n",
       "6       GRCh37:chr13:+:9783057:9783057/T/G             tcga-b6-a1kn-01a   \n",
       "7      GRCh37:chr3:+:68030078:68030078/C/T             tcga-b6-a1kn-01a   \n",
       "8      GRCh37:chr8:+:30665245:30665245/G/A             tcga-b6-a1kn-01a   \n",
       "9     GRCh37:chr19:+:43853546:43853546/C/T             tcga-b6-a1kn-01a   \n",
       "11  GRCh37:chr14:+:187130348:187130348/C/T             tcga-b6-a1kn-01a   \n",
       "12   GRCh37:chr1:+:111739850:111739850/G/A             tcga-b6-a1kn-01a   \n",
       "13    GRCh37:chr11:+:38390212:38390212/G/A             tcga-b6-a1kn-01a   \n",
       "15     GRCh37:chr5:+:31714438:31714438/C/A             tcga-b6-a1kn-01a   \n",
       "16  GRCh37:chr10:+:176812363:176812363/T/A             tcga-b6-a1kn-01a   \n",
       "17     GRCh37:chr8:+:39240627:39240627/T/C             tcga-b6-a1kn-01a   \n",
       "18     GRCh37:chr8:+:39261786:39261786/G/A             tcga-b6-a1kn-01a   \n",
       "19  GRCh37:chr20:+:153130895:153130895/G/A             tcga-b6-a1kn-01a   \n",
       "20     GRCh37:chr4:+:57604136:57604136/C/A             tcga-b6-a1kn-01a   \n",
       "22   GRCh37:chr6:+:100252738:100252738/A/C             tcga-b6-a1kn-01a   \n",
       "23     GRCh37:chr6:+:23811029:23811029/G/A             tcga-b6-a1kn-01a   \n",
       "25  GRCh37:chr14:+:146576496:146576496/T/G             tcga-b6-a1kn-01a   \n",
       "26  GRCh37:chr10:+:242038987:242038987/A/C             tcga-b6-a1kn-01a   \n",
       "27  GRCh37:chr17:+:100679631:100679631/G/A             tcga-b6-a1kn-01a   \n",
       "28    GRCh37:chr12:+:38083918:38083918/G/A             tcga-b6-a1kn-01a   \n",
       "29    GRCh37:chr10:+:73438003:73438003/C/A             tcga-b6-a1kn-01a   \n",
       "30    GRCh37:chr16:+:86181085:86181085/G/T             tcga-b6-a1kn-01a   \n",
       "31   GRCh37:chr1:+:205275345:205275345/C/A             tcga-b6-a1kn-01a   \n",
       "32       GRCh37:chr9:+:9967532:9967532/C/T             tcga-b6-a1kn-01a   \n",
       "37     GRCh37:chr7:+:31097783:31097783/G/A             tcga-b6-a1kn-01a   \n",
       "39   GRCh37:chr1:+:100573002:100573002/C/A             tcga-b6-a1kn-01a   \n",
       "40   GRCh37:chr4:+:100732846:100732846/A/T             tcga-b6-a1kn-01a   \n",
       "41     GRCh37:chr4:+:11338943:11338943/G/T             tcga-b6-a1kn-01a   \n",
       "42    GRCh37:chr14:+:38029467:38029467/T/C             tcga-b6-a1kn-01a   \n",
       "43     GRCh37:chr5:+:75880582:75880582/T/G             tcga-b6-a1kn-01a   \n",
       "44  GRCh37:chr15:+:140022616:140022616/A/C             tcga-b6-a1kn-01a   \n",
       "46       GRCh37:chr8:+:7577144:7577144/A/G             tcga-b6-a1kn-01a   \n",
       "47     GRCh37:chr5:+:99890758:99890758/T/C             tcga-b6-a1kn-01a   \n",
       "48    GRCh37:chr18:+:67578547:67578547/T/G             tcga-b6-a1kn-01a   \n",
       "49  GRCh37:chr13:+:101375072:101375072/G/C             tcga-b6-a1kn-01a   \n",
       "50    GRCh37:chr15:+:71739680:71739680/G/T             tcga-b6-a1kn-01a   \n",
       "51       GRCh37:chr9:+:9770066:9770066/T/C             tcga-b6-a1kn-01a   \n",
       "\n",
       "                                variant-id variant-ref-allele  \\\n",
       "0      GRCh37:chr3:+:10500146:10500146/C/T                  C   \n",
       "1      GRCh37:chr7:+:25228619:25228619/G/T                  G   \n",
       "3     GRCh37:chr11:+:41032794:41032794/C/T                  C   \n",
       "4      GRCh37:chr4:+:24989509:24989509/G/A                  G   \n",
       "5   GRCh37:chr20:+:129155089:129155089/C/A                  C   \n",
       "6       GRCh37:chr13:+:9783057:9783057/T/G                  T   \n",
       "7      GRCh37:chr3:+:68030078:68030078/C/T                  C   \n",
       "8      GRCh37:chr8:+:30665245:30665245/G/A                  G   \n",
       "9     GRCh37:chr19:+:43853546:43853546/C/T                  C   \n",
       "11  GRCh37:chr14:+:187130348:187130348/C/T                  C   \n",
       "12   GRCh37:chr1:+:111739850:111739850/G/A                  G   \n",
       "13    GRCh37:chr11:+:38390212:38390212/G/A                  G   \n",
       "15     GRCh37:chr5:+:31714438:31714438/C/A                  C   \n",
       "16  GRCh37:chr10:+:176812363:176812363/T/A                  T   \n",
       "17     GRCh37:chr8:+:39240627:39240627/T/C                  T   \n",
       "18     GRCh37:chr8:+:39261786:39261786/G/A                  G   \n",
       "19  GRCh37:chr20:+:153130895:153130895/G/A                  G   \n",
       "20     GRCh37:chr4:+:57604136:57604136/C/A                  C   \n",
       "22   GRCh37:chr6:+:100252738:100252738/A/C                  A   \n",
       "23     GRCh37:chr6:+:23811029:23811029/G/A                  G   \n",
       "25  GRCh37:chr14:+:146576496:146576496/T/G                  T   \n",
       "26  GRCh37:chr10:+:242038987:242038987/A/C                  A   \n",
       "27  GRCh37:chr17:+:100679631:100679631/G/A                  G   \n",
       "28    GRCh37:chr12:+:38083918:38083918/G/A                  G   \n",
       "29    GRCh37:chr10:+:73438003:73438003/C/A                  C   \n",
       "30    GRCh37:chr16:+:86181085:86181085/G/T                  G   \n",
       "31   GRCh37:chr1:+:205275345:205275345/C/A                  C   \n",
       "32       GRCh37:chr9:+:9967532:9967532/C/T                  C   \n",
       "37     GRCh37:chr7:+:31097783:31097783/G/A                  G   \n",
       "39   GRCh37:chr1:+:100573002:100573002/C/A                  C   \n",
       "40   GRCh37:chr4:+:100732846:100732846/A/T                  A   \n",
       "41     GRCh37:chr4:+:11338943:11338943/G/T                  G   \n",
       "42    GRCh37:chr14:+:38029467:38029467/T/C                  T   \n",
       "43     GRCh37:chr5:+:75880582:75880582/T/G                  T   \n",
       "44  GRCh37:chr15:+:140022616:140022616/A/C                  A   \n",
       "46       GRCh37:chr8:+:7577144:7577144/A/G                  A   \n",
       "47     GRCh37:chr5:+:99890758:99890758/T/C                  T   \n",
       "48    GRCh37:chr18:+:67578547:67578547/T/G                  T   \n",
       "49  GRCh37:chr13:+:101375072:101375072/G/C                  G   \n",
       "50    GRCh37:chr15:+:71739680:71739680/G/T                  G   \n",
       "51       GRCh37:chr9:+:9770066:9770066/T/C                  T   \n",
       "\n",
       "   variant-alt-allele variant-gene-gene-hgnc-symbol  \\\n",
       "0                   T                        OR5D14   \n",
       "1                   T                       ZCCHC12   \n",
       "3                   T                       ATP13A2   \n",
       "4                   A                          CPN2   \n",
       "5                   A                       CNTNAP5   \n",
       "6                   G                         MST1R   \n",
       "7                   T                         CERS1   \n",
       "8                   A                      APOBEC3B   \n",
       "9                   T                         GPAT2   \n",
       "11                  T                    KRTAP10-10   \n",
       "12                  A                       SLC41A3   \n",
       "13                  A                         ASCC1   \n",
       "15                  A                         PEBP4   \n",
       "16                  A                          GYS2   \n",
       "17                  C                          TPH2   \n",
       "18                  A                         CRLF3   \n",
       "19                  A                         SIDT2   \n",
       "20                  A                        PHLPP2   \n",
       "22                  C                           RP9   \n",
       "23                  A                         ASB14   \n",
       "25                  G                        C2CD4C   \n",
       "26                  C                        ACOT11   \n",
       "27                  A                           OGN   \n",
       "28                  A                      SLC25A53   \n",
       "29                  A                          HCN4   \n",
       "30                  T                        ZNF112   \n",
       "31                  A                          UPRT   \n",
       "32                  T                       WFDC10B   \n",
       "37                  A                       CARNMT1   \n",
       "39                  A                      ATP6V1E2   \n",
       "40                  T                         WDTC1   \n",
       "41                  T                       ABHD17C   \n",
       "42                  C                        FBXO46   \n",
       "43                  G                        TANGO6   \n",
       "44                  C                       CBFA2T3   \n",
       "46                  G                        KRBOX4   \n",
       "47                  C                         CCND1   \n",
       "48                  G                         DMBX1   \n",
       "49                  C                          COPE   \n",
       "50                  T                        NFATC4   \n",
       "51                  C                        MMADHC   \n",
       "\n",
       "              variant-classification  \n",
       "0   :variant.classification/missense  \n",
       "1   :variant.classification/missense  \n",
       "3   :variant.classification/missense  \n",
       "4   :variant.classification/missense  \n",
       "5   :variant.classification/missense  \n",
       "6   :variant.classification/missense  \n",
       "7   :variant.classification/missense  \n",
       "8   :variant.classification/missense  \n",
       "9   :variant.classification/missense  \n",
       "11  :variant.classification/missense  \n",
       "12  :variant.classification/missense  \n",
       "13  :variant.classification/missense  \n",
       "15  :variant.classification/missense  \n",
       "16  :variant.classification/missense  \n",
       "17  :variant.classification/missense  \n",
       "18  :variant.classification/missense  \n",
       "19  :variant.classification/missense  \n",
       "20  :variant.classification/missense  \n",
       "22  :variant.classification/missense  \n",
       "23  :variant.classification/missense  \n",
       "25  :variant.classification/missense  \n",
       "26  :variant.classification/missense  \n",
       "27  :variant.classification/missense  \n",
       "28  :variant.classification/missense  \n",
       "29  :variant.classification/missense  \n",
       "30  :variant.classification/missense  \n",
       "31  :variant.classification/missense  \n",
       "32  :variant.classification/missense  \n",
       "37  :variant.classification/missense  \n",
       "39  :variant.classification/missense  \n",
       "40  :variant.classification/missense  \n",
       "41  :variant.classification/missense  \n",
       "42  :variant.classification/missense  \n",
       "43  :variant.classification/missense  \n",
       "44  :variant.classification/missense  \n",
       "46  :variant.classification/missense  \n",
       "47  :variant.classification/missense  \n",
       "48  :variant.classification/missense  \n",
       "49  :variant.classification/missense  \n",
       "50  :variant.classification/missense  \n",
       "51  :variant.classification/missense  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>measurement-variant-variant-id</th>\n",
       "      <th>measurement-sample-sample-id</th>\n",
       "      <th>variant-id</th>\n",
       "      <th>variant-ref-allele</th>\n",
       "      <th>variant-alt-allele</th>\n",
       "      <th>variant-gene-gene-hgnc-symbol</th>\n",
       "      <th>variant-classification</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GRCh37:chr3:+:10500146:10500146/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr3:+:10500146:10500146/C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>OR5D14</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GRCh37:chr7:+:25228619:25228619/G/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr7:+:25228619:25228619/G/T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ZCCHC12</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GRCh37:chr11:+:41032794:41032794/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr11:+:41032794:41032794/C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ATP13A2</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GRCh37:chr4:+:24989509:24989509/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr4:+:24989509:24989509/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CPN2</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GRCh37:chr20:+:129155089:129155089/C/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr20:+:129155089:129155089/C/A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>CNTNAP5</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GRCh37:chr13:+:9783057:9783057/T/G</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr13:+:9783057:9783057/T/G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>MST1R</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GRCh37:chr3:+:68030078:68030078/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr3:+:68030078:68030078/C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CERS1</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GRCh37:chr8:+:30665245:30665245/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr8:+:30665245:30665245/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>APOBEC3B</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GRCh37:chr19:+:43853546:43853546/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr19:+:43853546:43853546/C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>GPAT2</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GRCh37:chr14:+:187130348:187130348/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr14:+:187130348:187130348/C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>KRTAP10-10</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GRCh37:chr1:+:111739850:111739850/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr1:+:111739850:111739850/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SLC41A3</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GRCh37:chr11:+:38390212:38390212/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr11:+:38390212:38390212/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ASCC1</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GRCh37:chr5:+:31714438:31714438/C/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr5:+:31714438:31714438/C/A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>PEBP4</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GRCh37:chr10:+:176812363:176812363/T/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr10:+:176812363:176812363/T/A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>GYS2</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GRCh37:chr8:+:39240627:39240627/T/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr8:+:39240627:39240627/T/C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>TPH2</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GRCh37:chr8:+:39261786:39261786/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr8:+:39261786:39261786/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CRLF3</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GRCh37:chr20:+:153130895:153130895/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr20:+:153130895:153130895/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SIDT2</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GRCh37:chr4:+:57604136:57604136/C/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr4:+:57604136:57604136/C/A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>PHLPP2</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GRCh37:chr6:+:100252738:100252738/A/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr6:+:100252738:100252738/A/C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>RP9</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GRCh37:chr6:+:23811029:23811029/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr6:+:23811029:23811029/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ASB14</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GRCh37:chr14:+:146576496:146576496/T/G</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr14:+:146576496:146576496/T/G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>C2CD4C</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>GRCh37:chr10:+:242038987:242038987/A/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr10:+:242038987:242038987/A/C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>ACOT11</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GRCh37:chr17:+:100679631:100679631/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr17:+:100679631:100679631/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>OGN</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GRCh37:chr12:+:38083918:38083918/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr12:+:38083918:38083918/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SLC25A53</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GRCh37:chr10:+:73438003:73438003/C/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr10:+:73438003:73438003/C/A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>HCN4</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>GRCh37:chr16:+:86181085:86181085/G/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr16:+:86181085:86181085/G/T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ZNF112</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GRCh37:chr1:+:205275345:205275345/C/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr1:+:205275345:205275345/C/A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>UPRT</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GRCh37:chr9:+:9967532:9967532/C/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr9:+:9967532:9967532/C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>WFDC10B</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>GRCh37:chr7:+:31097783:31097783/G/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr7:+:31097783:31097783/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CARNMT1</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>GRCh37:chr1:+:100573002:100573002/C/A</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr1:+:100573002:100573002/C/A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>ATP6V1E2</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>GRCh37:chr4:+:100732846:100732846/A/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr4:+:100732846:100732846/A/T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>WDTC1</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>GRCh37:chr4:+:11338943:11338943/G/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr4:+:11338943:11338943/G/T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ABHD17C</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>GRCh37:chr14:+:38029467:38029467/T/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr14:+:38029467:38029467/T/C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>FBXO46</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>GRCh37:chr5:+:75880582:75880582/T/G</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr5:+:75880582:75880582/T/G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>TANGO6</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>GRCh37:chr15:+:140022616:140022616/A/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr15:+:140022616:140022616/A/C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>CBFA2T3</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>GRCh37:chr8:+:7577144:7577144/A/G</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr8:+:7577144:7577144/A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>KRBOX4</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>GRCh37:chr5:+:99890758:99890758/T/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr5:+:99890758:99890758/T/C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>CCND1</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>GRCh37:chr18:+:67578547:67578547/T/G</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr18:+:67578547:67578547/T/G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>DMBX1</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>GRCh37:chr13:+:101375072:101375072/G/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr13:+:101375072:101375072/G/C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>COPE</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>GRCh37:chr15:+:71739680:71739680/G/T</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr15:+:71739680:71739680/G/T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NFATC4</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>GRCh37:chr9:+:9770066:9770066/T/C</td>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>GRCh37:chr9:+:9770066:9770066/T/C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>MMADHC</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 45
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
