{
 "cells": [
  {
   "cell_type": "code",
   "id": "initial_id",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-06-18T20:35:12.308799Z",
     "start_time": "2025-06-18T20:35:12.079276Z"
    }
   },
   "source": [
    "%%capture\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "import pandas as pd\n",
    "import patternq.query as pqq\n",
    "import patternq.dataset as pqd\n",
    "import patternq.reference as pqr"
   ],
   "outputs": [],
   "execution_count": 1
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": [
    "## PatternQ Query Wrappers Example\n",
    "\n",
    "This notebook shows an example of access data in the Pattern Data Commons,\n",
    "using the query wrapping convience routines in `patternq.dataset` and `patternq.reference`."
   ],
   "id": "78177f0502ec1e1c"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-06-18T20:35:12.326027Z",
     "start_time": "2025-06-18T20:35:12.312446Z"
    }
   },
   "cell_type": "code",
   "source": [
    "%%capture\n",
    "import os\n",
    "os.environ\n",
    "# confirm PATTERNQ_API_KEY is set to your API key here."
   ],
   "id": "3e9e5957368f074e",
   "outputs": [],
   "execution_count": 2
  },
  {
   "cell_type": "code",
   "id": "79f2439641b3d0d6",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-06-18T20:35:13.140010Z",
     "start_time": "2025-06-18T20:35:12.399750Z"
    }
   },
   "source": [
    "subjects = pqd.all_subjects('tcga-brca', db_name='tcga-brca')\n",
    "subjects"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                          subject-HLA-A-type  \\\n",
       "0                           [hla_a_68_01_02]   \n",
       "1           [hla_a_25_01_01, hla_a_31_01_02]   \n",
       "2                        [hla_a_03_01_01_01]   \n",
       "3                           [hla_a_26_01_03]   \n",
       "4                        [hla_a_02_01_01_01]   \n",
       "...                                      ...   \n",
       "1093     [hla_a_01_01_01_01, hla_a_26_01_01]   \n",
       "1094  [hla_a_01_01_01_01, hla_a_24_02_01_01]   \n",
       "1095                     [hla_a_02_01_01_01]   \n",
       "1096     [hla_a_31_01_02, hla_a_80_01_01_01]   \n",
       "1097                                     NaN   \n",
       "\n",
       "                                        subject-uid  \\\n",
       "0     [tcga-brca, 48958e0a033b76dd2cde746ec340d7b8]   \n",
       "1     [tcga-brca, cdff6ccae7b3a01e06583b6eb6935647]   \n",
       "2     [tcga-brca, 58de8decc5806306e122d7d2e0361077]   \n",
       "3     [tcga-brca, e33c10ac862066f01799593f93226d63]   \n",
       "4     [tcga-brca, 33c8d0592a63e6d929493c6fee83049a]   \n",
       "...                                             ...   \n",
       "1093  [tcga-brca, 28355fea1611cf76788e36f7d8d1e63b]   \n",
       "1094  [tcga-brca, b49d76c1ed3ce2f807c30aca593700e5]   \n",
       "1095  [tcga-brca, 679b389e478544f624e010f2fba4951c]   \n",
       "1096  [tcga-brca, ac707e57a213016c5f7f1a643c938bc0]   \n",
       "1097  [tcga-brca, bd92ea29e8454b46c4e0c940f391887a]   \n",
       "\n",
       "                        subject-race   subject-freetext-disease  \\\n",
       "0                :subject.race/white  Invasive breast carcinoma   \n",
       "1                :subject.race/white  Invasive breast carcinoma   \n",
       "2                :subject.race/white  Invasive breast carcinoma   \n",
       "3                :subject.race/white  Invasive breast carcinoma   \n",
       "4                :subject.race/white  Invasive breast carcinoma   \n",
       "...                              ...                        ...   \n",
       "1093             :subject.race/white  Invasive breast carcinoma   \n",
       "1094                             NaN  Invasive breast carcinoma   \n",
       "1095             :subject.race/white  Invasive breast carcinoma   \n",
       "1096  :subject.race/african-american  Invasive breast carcinoma   \n",
       "1097                             NaN  Invasive breast carcinoma   \n",
       "\n",
       "                       subject-HLA-B-type           db-id  subject-age  \\\n",
       "0     [hla_b_40_01_02, hla_b_44_02_01_03]  17592187509997         50.0   \n",
       "1        [hla_b_18_01_01_02, hla_b_51_08]  17592187510001         68.0   \n",
       "2                        [hla_b_07_02_01]  17592187510005         35.0   \n",
       "3        [hla_b_38_01_01, hla_b_40_02_01]  17592187510009         55.0   \n",
       "4     [hla_b_13_02_01, hla_b_18_01_01_02]  17592187510013         83.0   \n",
       "...                                   ...             ...          ...   \n",
       "1093     [hla_b_07_02_01, hla_b_08_01_07]  17592187514126         44.0   \n",
       "1094     [hla_b_08_01_01, hla_b_44_05_01]  17592187514130         63.0   \n",
       "1095  [hla_b_07_02_01, hla_b_35_01_01_02]  17592187514133         53.0   \n",
       "1096     [hla_b_27_05_02, hla_b_40_01_02]  17592187517522          NaN   \n",
       "1097                                  NaN  17592234961832          NaN   \n",
       "\n",
       "              subject-sex                      subject-HLA-C-type  \\\n",
       "0     :subject.sex/female     [hla_c_03_04_01_01, hla_c_07_04_01]   \n",
       "1     :subject.sex/female     [hla_c_12_03_01_01, hla_c_16_02_01]   \n",
       "2     :subject.sex/female                     [hla_c_07_02_01_03]   \n",
       "3     :subject.sex/female     [hla_c_02_02_02, hla_c_12_03_01_01]   \n",
       "4     :subject.sex/female  [hla_c_06_02_01_01, hla_c_07_01_01_01]   \n",
       "...                   ...                                     ...   \n",
       "1093  :subject.sex/female  [hla_c_07_01_01_01, hla_c_07_02_01_03]   \n",
       "1094  :subject.sex/female     [hla_c_02_02_02, hla_c_07_01_01_01]   \n",
       "1095  :subject.sex/female  [hla_c_04_01_01_01, hla_c_07_02_01_03]   \n",
       "1096  :subject.sex/female  [hla_c_03_04_01_01, hla_c_06_02_01_01]   \n",
       "1097                  NaN                                     NaN   \n",
       "\n",
       "     subject-dead    subject-id                subject-ethnicity  \n",
       "0            True  TCGA-AR-A1AR                              NaN  \n",
       "1            True  TCGA-BH-A1EO                              NaN  \n",
       "2            True  TCGA-BH-A1ES                              NaN  \n",
       "3            True  TCGA-BH-A1ET                              NaN  \n",
       "4            True  TCGA-BH-A1EU                              NaN  \n",
       "...           ...           ...                              ...  \n",
       "1093        False  TCGA-E2-A1B6                              NaN  \n",
       "1094        False  TCGA-E2-A1BC                              NaN  \n",
       "1095        False  TCGA-E2-A1BD                              NaN  \n",
       "1096        False  TCGA-OL-A66H  :subject.ethnicity/not-hispanic  \n",
       "1097          NaN  TCGA-BH-A0B2                              NaN  \n",
       "\n",
       "[1098 rows x 12 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject-HLA-A-type</th>\n",
       "      <th>subject-uid</th>\n",
       "      <th>subject-race</th>\n",
       "      <th>subject-freetext-disease</th>\n",
       "      <th>subject-HLA-B-type</th>\n",
       "      <th>db-id</th>\n",
       "      <th>subject-age</th>\n",
       "      <th>subject-sex</th>\n",
       "      <th>subject-HLA-C-type</th>\n",
       "      <th>subject-dead</th>\n",
       "      <th>subject-id</th>\n",
       "      <th>subject-ethnicity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>[hla_a_68_01_02]</td>\n",
       "      <td>[tcga-brca, 48958e0a033b76dd2cde746ec340d7b8]</td>\n",
       "      <td>:subject.race/white</td>\n",
       "      <td>Invasive breast carcinoma</td>\n",
       "      <td>[hla_b_40_01_02, hla_b_44_02_01_03]</td>\n",
       "      <td>17592187509997</td>\n",
       "      <td>50.0</td>\n",
       "      <td>:subject.sex/female</td>\n",
       "      <td>[hla_c_03_04_01_01, hla_c_07_04_01]</td>\n",
       "      <td>True</td>\n",
       "      <td>TCGA-AR-A1AR</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>[hla_a_25_01_01, hla_a_31_01_02]</td>\n",
       "      <td>[tcga-brca, cdff6ccae7b3a01e06583b6eb6935647]</td>\n",
       "      <td>:subject.race/white</td>\n",
       "      <td>Invasive breast carcinoma</td>\n",
       "      <td>[hla_b_18_01_01_02, hla_b_51_08]</td>\n",
       "      <td>17592187510001</td>\n",
       "      <td>68.0</td>\n",
       "      <td>:subject.sex/female</td>\n",
       "      <td>[hla_c_12_03_01_01, hla_c_16_02_01]</td>\n",
       "      <td>True</td>\n",
       "      <td>TCGA-BH-A1EO</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>[hla_a_03_01_01_01]</td>\n",
       "      <td>[tcga-brca, 58de8decc5806306e122d7d2e0361077]</td>\n",
       "      <td>:subject.race/white</td>\n",
       "      <td>Invasive breast carcinoma</td>\n",
       "      <td>[hla_b_07_02_01]</td>\n",
       "      <td>17592187510005</td>\n",
       "      <td>35.0</td>\n",
       "      <td>:subject.sex/female</td>\n",
       "      <td>[hla_c_07_02_01_03]</td>\n",
       "      <td>True</td>\n",
       "      <td>TCGA-BH-A1ES</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>[hla_a_26_01_03]</td>\n",
       "      <td>[tcga-brca, e33c10ac862066f01799593f93226d63]</td>\n",
       "      <td>:subject.race/white</td>\n",
       "      <td>Invasive breast carcinoma</td>\n",
       "      <td>[hla_b_38_01_01, hla_b_40_02_01]</td>\n",
       "      <td>17592187510009</td>\n",
       "      <td>55.0</td>\n",
       "      <td>:subject.sex/female</td>\n",
       "      <td>[hla_c_02_02_02, hla_c_12_03_01_01]</td>\n",
       "      <td>True</td>\n",
       "      <td>TCGA-BH-A1ET</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>[hla_a_02_01_01_01]</td>\n",
       "      <td>[tcga-brca, 33c8d0592a63e6d929493c6fee83049a]</td>\n",
       "      <td>:subject.race/white</td>\n",
       "      <td>Invasive breast carcinoma</td>\n",
       "      <td>[hla_b_13_02_01, hla_b_18_01_01_02]</td>\n",
       "      <td>17592187510013</td>\n",
       "      <td>83.0</td>\n",
       "      <td>:subject.sex/female</td>\n",
       "      <td>[hla_c_06_02_01_01, hla_c_07_01_01_01]</td>\n",
       "      <td>True</td>\n",
       "      <td>TCGA-BH-A1EU</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1093</th>\n",
       "      <td>[hla_a_01_01_01_01, hla_a_26_01_01]</td>\n",
       "      <td>[tcga-brca, 28355fea1611cf76788e36f7d8d1e63b]</td>\n",
       "      <td>:subject.race/white</td>\n",
       "      <td>Invasive breast carcinoma</td>\n",
       "      <td>[hla_b_07_02_01, hla_b_08_01_07]</td>\n",
       "      <td>17592187514126</td>\n",
       "      <td>44.0</td>\n",
       "      <td>:subject.sex/female</td>\n",
       "      <td>[hla_c_07_01_01_01, hla_c_07_02_01_03]</td>\n",
       "      <td>False</td>\n",
       "      <td>TCGA-E2-A1B6</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1094</th>\n",
       "      <td>[hla_a_01_01_01_01, hla_a_24_02_01_01]</td>\n",
       "      <td>[tcga-brca, b49d76c1ed3ce2f807c30aca593700e5]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Invasive breast carcinoma</td>\n",
       "      <td>[hla_b_08_01_01, hla_b_44_05_01]</td>\n",
       "      <td>17592187514130</td>\n",
       "      <td>63.0</td>\n",
       "      <td>:subject.sex/female</td>\n",
       "      <td>[hla_c_02_02_02, hla_c_07_01_01_01]</td>\n",
       "      <td>False</td>\n",
       "      <td>TCGA-E2-A1BC</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1095</th>\n",
       "      <td>[hla_a_02_01_01_01]</td>\n",
       "      <td>[tcga-brca, 679b389e478544f624e010f2fba4951c]</td>\n",
       "      <td>:subject.race/white</td>\n",
       "      <td>Invasive breast carcinoma</td>\n",
       "      <td>[hla_b_07_02_01, hla_b_35_01_01_02]</td>\n",
       "      <td>17592187514133</td>\n",
       "      <td>53.0</td>\n",
       "      <td>:subject.sex/female</td>\n",
       "      <td>[hla_c_04_01_01_01, hla_c_07_02_01_03]</td>\n",
       "      <td>False</td>\n",
       "      <td>TCGA-E2-A1BD</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>[hla_a_31_01_02, hla_a_80_01_01_01]</td>\n",
       "      <td>[tcga-brca, ac707e57a213016c5f7f1a643c938bc0]</td>\n",
       "      <td>:subject.race/african-american</td>\n",
       "      <td>Invasive breast carcinoma</td>\n",
       "      <td>[hla_b_27_05_02, hla_b_40_01_02]</td>\n",
       "      <td>17592187517522</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:subject.sex/female</td>\n",
       "      <td>[hla_c_03_04_01_01, hla_c_06_02_01_01]</td>\n",
       "      <td>False</td>\n",
       "      <td>TCGA-OL-A66H</td>\n",
       "      <td>:subject.ethnicity/not-hispanic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>NaN</td>\n",
       "      <td>[tcga-brca, bd92ea29e8454b46c4e0c940f391887a]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Invasive breast carcinoma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17592234961832</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-BH-A0B2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1098 rows × 12 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 3
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": [
    "Because a patient can have multiple HLA A, B, or C types, we explode the column here\n",
    "(create multiple rows when we have multiple values in a list in a single cell in the DataFrame)."
   ],
   "id": "7202b8a6566a87f6"
  },
  {
   "cell_type": "code",
   "id": "eefb2df20c58950c",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-06-18T20:35:13.188697Z",
     "start_time": "2025-06-18T20:35:13.172139Z"
    }
   },
   "source": [
    "by_hla = subjects[['subject-id', 'subject-HLA-A-type', 'subject-HLA-B-type', 'subject-HLA-C-type']].explode(column='subject-HLA-A-type')\n",
    "by_hla = by_hla.explode(column='subject-HLA-B-type')\n",
    "by_hla = by_hla.explode(column='subject-HLA-C-type')\n",
    "by_hla"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        subject-id subject-HLA-A-type subject-HLA-B-type subject-HLA-C-type\n",
       "0     TCGA-AR-A1AR     hla_a_68_01_02     hla_b_40_01_02  hla_c_03_04_01_01\n",
       "0     TCGA-AR-A1AR     hla_a_68_01_02     hla_b_40_01_02     hla_c_07_04_01\n",
       "0     TCGA-AR-A1AR     hla_a_68_01_02  hla_b_44_02_01_03  hla_c_03_04_01_01\n",
       "0     TCGA-AR-A1AR     hla_a_68_01_02  hla_b_44_02_01_03     hla_c_07_04_01\n",
       "1     TCGA-BH-A1EO     hla_a_25_01_01  hla_b_18_01_01_02  hla_c_12_03_01_01\n",
       "...            ...                ...                ...                ...\n",
       "1096  TCGA-OL-A66H  hla_a_80_01_01_01     hla_b_27_05_02  hla_c_03_04_01_01\n",
       "1096  TCGA-OL-A66H  hla_a_80_01_01_01     hla_b_27_05_02  hla_c_06_02_01_01\n",
       "1096  TCGA-OL-A66H  hla_a_80_01_01_01     hla_b_40_01_02  hla_c_03_04_01_01\n",
       "1096  TCGA-OL-A66H  hla_a_80_01_01_01     hla_b_40_01_02  hla_c_06_02_01_01\n",
       "1097  TCGA-BH-A0B2                NaN                NaN                NaN\n",
       "\n",
       "[7217 rows x 4 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject-id</th>\n",
       "      <th>subject-HLA-A-type</th>\n",
       "      <th>subject-HLA-B-type</th>\n",
       "      <th>subject-HLA-C-type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-AR-A1AR</td>\n",
       "      <td>hla_a_68_01_02</td>\n",
       "      <td>hla_b_40_01_02</td>\n",
       "      <td>hla_c_03_04_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-AR-A1AR</td>\n",
       "      <td>hla_a_68_01_02</td>\n",
       "      <td>hla_b_40_01_02</td>\n",
       "      <td>hla_c_07_04_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-AR-A1AR</td>\n",
       "      <td>hla_a_68_01_02</td>\n",
       "      <td>hla_b_44_02_01_03</td>\n",
       "      <td>hla_c_03_04_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-AR-A1AR</td>\n",
       "      <td>hla_a_68_01_02</td>\n",
       "      <td>hla_b_44_02_01_03</td>\n",
       "      <td>hla_c_07_04_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-BH-A1EO</td>\n",
       "      <td>hla_a_25_01_01</td>\n",
       "      <td>hla_b_18_01_01_02</td>\n",
       "      <td>hla_c_12_03_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>TCGA-OL-A66H</td>\n",
       "      <td>hla_a_80_01_01_01</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_03_04_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>TCGA-OL-A66H</td>\n",
       "      <td>hla_a_80_01_01_01</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_06_02_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>TCGA-OL-A66H</td>\n",
       "      <td>hla_a_80_01_01_01</td>\n",
       "      <td>hla_b_40_01_02</td>\n",
       "      <td>hla_c_03_04_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>TCGA-OL-A66H</td>\n",
       "      <td>hla_a_80_01_01_01</td>\n",
       "      <td>hla_b_40_01_02</td>\n",
       "      <td>hla_c_06_02_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>TCGA-BH-A0B2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7217 rows × 4 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 4
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "Here we use the HLA B data to limit follow-on queries to patients who are HLA B27 positive, a marker common in autoimmune disease.",
   "id": "87f54b9a5c25638c"
  },
  {
   "cell_type": "code",
   "id": "6c6ffaa2d8cc0d78",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-06-18T20:35:13.254048Z",
     "start_time": "2025-06-18T20:35:13.244391Z"
    }
   },
   "source": [
    "by_hla['subject-HLA-B-type'].value_counts()"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "subject-HLA-B-type\n",
       "hla_b_07_02_01       841\n",
       "hla_b_08_01_01       508\n",
       "hla_b_15_01_01_01    423\n",
       "hla_b_44_02_01_01    393\n",
       "hla_b_35_01_01_02    332\n",
       "                    ... \n",
       "hla_b_51_07_02         2\n",
       "hla_b_39_01_03         1\n",
       "hla_b_44_02_15         1\n",
       "hla_b_14_03            1\n",
       "hla_b_07_02_06         1\n",
       "Name: count, Length: 102, dtype: int64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 5
  },
  {
   "cell_type": "code",
   "id": "f8982b790a03f6ed",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-06-18T20:35:13.348395Z",
     "start_time": "2025-06-18T20:35:13.335850Z"
    }
   },
   "source": [
    "b27_matches = by_hla['subject-HLA-B-type'].astype('string').str.contains(\"b_27\")\n",
    "b27_matches.value_counts()\n",
    "b27_sub = by_hla[b27_matches]\n",
    "b27_sub"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        subject-id subject-HLA-A-type subject-HLA-B-type subject-HLA-C-type\n",
       "15    TCGA-E2-A15I     hla_a_03_01_19     hla_b_27_05_02     hla_c_01_02_01\n",
       "15    TCGA-E2-A15I     hla_a_03_01_19     hla_b_27_05_02     hla_c_15_02_01\n",
       "15    TCGA-E2-A15I     hla_a_26_01_03     hla_b_27_05_02     hla_c_01_02_01\n",
       "15    TCGA-E2-A15I     hla_a_26_01_03     hla_b_27_05_02     hla_c_15_02_01\n",
       "33    TCGA-B6-A1KN  hla_a_03_01_01_01     hla_b_27_05_02     hla_c_02_02_02\n",
       "...            ...                ...                ...                ...\n",
       "1064  TCGA-AR-A1AQ     hla_a_02_06_05     hla_b_27_02_01     hla_c_16_01_01\n",
       "1096  TCGA-OL-A66H     hla_a_31_01_02     hla_b_27_05_02  hla_c_03_04_01_01\n",
       "1096  TCGA-OL-A66H     hla_a_31_01_02     hla_b_27_05_02  hla_c_06_02_01_01\n",
       "1096  TCGA-OL-A66H  hla_a_80_01_01_01     hla_b_27_05_02  hla_c_03_04_01_01\n",
       "1096  TCGA-OL-A66H  hla_a_80_01_01_01     hla_b_27_05_02  hla_c_06_02_01_01\n",
       "\n",
       "[297 rows x 4 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject-id</th>\n",
       "      <th>subject-HLA-A-type</th>\n",
       "      <th>subject-HLA-B-type</th>\n",
       "      <th>subject-HLA-C-type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TCGA-E2-A15I</td>\n",
       "      <td>hla_a_03_01_19</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_01_02_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TCGA-E2-A15I</td>\n",
       "      <td>hla_a_03_01_19</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_15_02_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TCGA-E2-A15I</td>\n",
       "      <td>hla_a_26_01_03</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_01_02_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TCGA-E2-A15I</td>\n",
       "      <td>hla_a_26_01_03</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_15_02_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>TCGA-B6-A1KN</td>\n",
       "      <td>hla_a_03_01_01_01</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_02_02_02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1064</th>\n",
       "      <td>TCGA-AR-A1AQ</td>\n",
       "      <td>hla_a_02_06_05</td>\n",
       "      <td>hla_b_27_02_01</td>\n",
       "      <td>hla_c_16_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>TCGA-OL-A66H</td>\n",
       "      <td>hla_a_31_01_02</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_03_04_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>TCGA-OL-A66H</td>\n",
       "      <td>hla_a_31_01_02</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_06_02_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>TCGA-OL-A66H</td>\n",
       "      <td>hla_a_80_01_01_01</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_03_04_01_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>TCGA-OL-A66H</td>\n",
       "      <td>hla_a_80_01_01_01</td>\n",
       "      <td>hla_b_27_05_02</td>\n",
       "      <td>hla_c_06_02_01_01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>297 rows × 4 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 6
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "Then, for patients in this group, we find which assays have been run on their samples.",
   "id": "63e66f1d8c5113de"
  },
  {
   "cell_type": "code",
   "id": "bb87ca446c002c8a",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-06-18T20:35:13.904814Z",
     "start_time": "2025-06-18T20:35:13.403754Z"
    }
   },
   "source": [
    "sub_ids = b27_sub['subject-id'].unique().tolist()\n",
    "assays_for_sub = pqd.assays_for_patients(sub_ids, db_name='tcga-brca')\n",
    "assays_for_sub"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "       subject-id         sample-id                       assay-tech  \\\n",
       "0    TCGA-A2-A0ER  tcga-a2-a0er-10a        :assay.technology/RNA-seq   \n",
       "1    TCGA-A2-A0ST  tcga-a2-a0st-10a        :assay.technology/RNA-seq   \n",
       "2    TCGA-A7-A13H  tcga-a7-a13h-01a      :assay.technology/SNP-array   \n",
       "3    TCGA-AN-A0AL  tcga-an-a0al-01a      :assay.technology/SNP-array   \n",
       "4    TCGA-A8-A099  tcga-a8-a099-01a            :assay.technology/WES   \n",
       "..            ...               ...                              ...   \n",
       "596  TCGA-A2-A04V  tcga-a2-a04v-01a        :assay.technology/RNA-seq   \n",
       "597  TCGA-AC-A3W6  tcga-ac-a3w6-01a            :assay.technology/WES   \n",
       "598  TCGA-OL-A66H  tcga-ol-a66h-01a  :assay.technology/miscellaneous   \n",
       "599  TCGA-E9-A1R6  tcga-e9-a1r6-01a        :assay.technology/RNA-seq   \n",
       "600  TCGA-AO-A0JD  tcga-ao-a0jd-01a            :assay.technology/WES   \n",
       "\n",
       "           assay-name       measurement-set-name  \n",
       "0        RNA-seq data  immune-cell-deconvolution  \n",
       "1        RNA-seq data  immune-cell-deconvolution  \n",
       "2    cnv and variants               snp CNV data  \n",
       "3    cnv and variants               snp CNV data  \n",
       "4                 WES         baseline mutations  \n",
       "..                ...                        ...  \n",
       "596      RNA-seq data                         gx  \n",
       "597               WES         baseline mutations  \n",
       "598      Tumor purity               tumor purity  \n",
       "599      RNA-seq data  immune-cell-deconvolution  \n",
       "600               WES         baseline mutations  \n",
       "\n",
       "[601 rows x 5 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject-id</th>\n",
       "      <th>sample-id</th>\n",
       "      <th>assay-tech</th>\n",
       "      <th>assay-name</th>\n",
       "      <th>measurement-set-name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-A2-A0ER</td>\n",
       "      <td>tcga-a2-a0er-10a</td>\n",
       "      <td>:assay.technology/RNA-seq</td>\n",
       "      <td>RNA-seq data</td>\n",
       "      <td>immune-cell-deconvolution</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-A2-A0ST</td>\n",
       "      <td>tcga-a2-a0st-10a</td>\n",
       "      <td>:assay.technology/RNA-seq</td>\n",
       "      <td>RNA-seq data</td>\n",
       "      <td>immune-cell-deconvolution</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-A7-A13H</td>\n",
       "      <td>tcga-a7-a13h-01a</td>\n",
       "      <td>:assay.technology/SNP-array</td>\n",
       "      <td>cnv and variants</td>\n",
       "      <td>snp CNV data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-AN-A0AL</td>\n",
       "      <td>tcga-an-a0al-01a</td>\n",
       "      <td>:assay.technology/SNP-array</td>\n",
       "      <td>cnv and variants</td>\n",
       "      <td>snp CNV data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-A8-A099</td>\n",
       "      <td>tcga-a8-a099-01a</td>\n",
       "      <td>:assay.technology/WES</td>\n",
       "      <td>WES</td>\n",
       "      <td>baseline mutations</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>596</th>\n",
       "      <td>TCGA-A2-A04V</td>\n",
       "      <td>tcga-a2-a04v-01a</td>\n",
       "      <td>:assay.technology/RNA-seq</td>\n",
       "      <td>RNA-seq data</td>\n",
       "      <td>gx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>597</th>\n",
       "      <td>TCGA-AC-A3W6</td>\n",
       "      <td>tcga-ac-a3w6-01a</td>\n",
       "      <td>:assay.technology/WES</td>\n",
       "      <td>WES</td>\n",
       "      <td>baseline mutations</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>598</th>\n",
       "      <td>TCGA-OL-A66H</td>\n",
       "      <td>tcga-ol-a66h-01a</td>\n",
       "      <td>:assay.technology/miscellaneous</td>\n",
       "      <td>Tumor purity</td>\n",
       "      <td>tumor purity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>599</th>\n",
       "      <td>TCGA-E9-A1R6</td>\n",
       "      <td>tcga-e9-a1r6-01a</td>\n",
       "      <td>:assay.technology/RNA-seq</td>\n",
       "      <td>RNA-seq data</td>\n",
       "      <td>immune-cell-deconvolution</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>600</th>\n",
       "      <td>TCGA-AO-A0JD</td>\n",
       "      <td>tcga-ao-a0jd-01a</td>\n",
       "      <td>:assay.technology/WES</td>\n",
       "      <td>WES</td>\n",
       "      <td>baseline mutations</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>601 rows × 5 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 7
  },
  {
   "cell_type": "code",
   "id": "f229b14d787db808",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-06-18T20:35:14.760458Z",
     "start_time": "2025-06-18T20:35:13.953735Z"
    }
   },
   "source": [
    "samples = pqd.samples(\"tcga-brca\", db_name='tcga-brca')\n",
    "samples"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "             sample-id sample-tumor-type                sample-specimen  \\\n",
       "0     tcga-a2-a0eu-01a            breast  :sample.specimen/fresh-frozen   \n",
       "1     tcga-a2-a0eu-10a               NaN  :sample.specimen/fresh-frozen   \n",
       "2     tcga-a7-a0d9-01a            breast  :sample.specimen/fresh-frozen   \n",
       "3     tcga-a7-a0d9-10a               NaN  :sample.specimen/fresh-frozen   \n",
       "4     tcga-ao-a0jf-01a            breast  :sample.specimen/fresh-frozen   \n",
       "...                ...               ...                            ...   \n",
       "2291  tcga-b6-a402-10a               NaN  :sample.specimen/fresh-frozen   \n",
       "2292  tcga-e9-a1ne-11a               NaN  :sample.specimen/fresh-frozen   \n",
       "2293  tcga-ol-a5da-10a               NaN  :sample.specimen/fresh-frozen   \n",
       "2294  tcga-bh-a5j0-10a               NaN  :sample.specimen/fresh-frozen   \n",
       "2295  tcga-bh-a0b1-01a            breast  :sample.specimen/fresh-frozen   \n",
       "\n",
       "     sample-metastasis                                     sample-uid  \\\n",
       "0                False  [tcga-brca, 1083184581b3c24ac85291d75a832b9a]   \n",
       "1                False  [tcga-brca, e96e9d5e9e41bba015ba8e4daefda967]   \n",
       "2                False  [tcga-brca, 8c1dbb6ba31a61b3b978a1f93dccb86c]   \n",
       "3                False  [tcga-brca, ba8add19b462211346377f6c7835cb6a]   \n",
       "4                False  [tcga-brca, 0bffbcada51f70f1978d207e22089d60]   \n",
       "...                ...                                            ...   \n",
       "2291             False  [tcga-brca, 9ede5e348369b0d13d4c7989eb565e3c]   \n",
       "2292             False  [tcga-brca, 85358f998e91a477db9bc868554ea46c]   \n",
       "2293             False  [tcga-brca, 7882611517385fe91546199a73b523c7]   \n",
       "2294             False  [tcga-brca, 934e529d67c8a4b6e7840615131ce315]   \n",
       "2295             False  [tcga-brca, 78ff94f90b8de61759b454cf020bd168]   \n",
       "\n",
       "               db-id sample-timepoint-timepoint-id sample-subject-subject-id  \\\n",
       "0     17592187518720            sample procurement              TCGA-A2-A0EU   \n",
       "1     17592187518803            sample procurement              TCGA-A2-A0EU   \n",
       "2     17592187518823            sample procurement              TCGA-A7-A0D9   \n",
       "3     17592187519258            sample procurement              TCGA-A7-A0D9   \n",
       "4     17592187519626            sample procurement              TCGA-AO-A0JF   \n",
       "...              ...                           ...                       ...   \n",
       "2291  17592188110342            sample procurement              TCGA-B6-A402   \n",
       "2292  17592188110466            sample procurement              TCGA-E9-A1NE   \n",
       "2293  17592188111439            sample procurement              TCGA-OL-A5DA   \n",
       "2294  17592188112629            sample procurement              TCGA-BH-A5J0   \n",
       "2295  17592188118826            sample procurement              TCGA-BH-A0B1   \n",
       "\n",
       "     sample-study-day-study-day-id  sample-type-db-id  \\\n",
       "0                               35     17592186045420   \n",
       "1                               35     17592186045421   \n",
       "2                               10     17592186045420   \n",
       "3                               10     17592186045421   \n",
       "4                               57     17592186045420   \n",
       "...                            ...                ...   \n",
       "2291                             0     17592186045421   \n",
       "2292                             0     17592186045421   \n",
       "2293                            33     17592186045421   \n",
       "2294                           104     17592186045421   \n",
       "2295                            44     17592186045420   \n",
       "\n",
       "     sample-gdc-anatomic-site-gdc-anatomic-site-name  \n",
       "0                                             Breast  \n",
       "1                                             Breast  \n",
       "2                                             Breast  \n",
       "3                                             Breast  \n",
       "4                                             Breast  \n",
       "...                                              ...  \n",
       "2291                                          Breast  \n",
       "2292                                          Breast  \n",
       "2293                                          Breast  \n",
       "2294                                          Breast  \n",
       "2295                                          Breast  \n",
       "\n",
       "[2296 rows x 11 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>sample-tumor-type</th>\n",
       "      <th>sample-specimen</th>\n",
       "      <th>sample-metastasis</th>\n",
       "      <th>sample-uid</th>\n",
       "      <th>db-id</th>\n",
       "      <th>sample-timepoint-timepoint-id</th>\n",
       "      <th>sample-subject-subject-id</th>\n",
       "      <th>sample-study-day-study-day-id</th>\n",
       "      <th>sample-type-db-id</th>\n",
       "      <th>sample-gdc-anatomic-site-gdc-anatomic-site-name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>tcga-a2-a0eu-01a</td>\n",
       "      <td>breast</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 1083184581b3c24ac85291d75a832b9a]</td>\n",
       "      <td>17592187518720</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-A2-A0EU</td>\n",
       "      <td>35</td>\n",
       "      <td>17592186045420</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>tcga-a2-a0eu-10a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, e96e9d5e9e41bba015ba8e4daefda967]</td>\n",
       "      <td>17592187518803</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-A2-A0EU</td>\n",
       "      <td>35</td>\n",
       "      <td>17592186045421</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>tcga-a7-a0d9-01a</td>\n",
       "      <td>breast</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 8c1dbb6ba31a61b3b978a1f93dccb86c]</td>\n",
       "      <td>17592187518823</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-A7-A0D9</td>\n",
       "      <td>10</td>\n",
       "      <td>17592186045420</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>tcga-a7-a0d9-10a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, ba8add19b462211346377f6c7835cb6a]</td>\n",
       "      <td>17592187519258</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-A7-A0D9</td>\n",
       "      <td>10</td>\n",
       "      <td>17592186045421</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>tcga-ao-a0jf-01a</td>\n",
       "      <td>breast</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 0bffbcada51f70f1978d207e22089d60]</td>\n",
       "      <td>17592187519626</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-AO-A0JF</td>\n",
       "      <td>57</td>\n",
       "      <td>17592186045420</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2291</th>\n",
       "      <td>tcga-b6-a402-10a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 9ede5e348369b0d13d4c7989eb565e3c]</td>\n",
       "      <td>17592188110342</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-B6-A402</td>\n",
       "      <td>0</td>\n",
       "      <td>17592186045421</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2292</th>\n",
       "      <td>tcga-e9-a1ne-11a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 85358f998e91a477db9bc868554ea46c]</td>\n",
       "      <td>17592188110466</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-E9-A1NE</td>\n",
       "      <td>0</td>\n",
       "      <td>17592186045421</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2293</th>\n",
       "      <td>tcga-ol-a5da-10a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 7882611517385fe91546199a73b523c7]</td>\n",
       "      <td>17592188111439</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-OL-A5DA</td>\n",
       "      <td>33</td>\n",
       "      <td>17592186045421</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2294</th>\n",
       "      <td>tcga-bh-a5j0-10a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 934e529d67c8a4b6e7840615131ce315]</td>\n",
       "      <td>17592188112629</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-BH-A5J0</td>\n",
       "      <td>104</td>\n",
       "      <td>17592186045421</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2295</th>\n",
       "      <td>tcga-bh-a0b1-01a</td>\n",
       "      <td>breast</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 78ff94f90b8de61759b454cf020bd168]</td>\n",
       "      <td>17592188118826</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-BH-A0B1</td>\n",
       "      <td>44</td>\n",
       "      <td>17592186045420</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2296 rows × 11 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 8
  },
  {
   "cell_type": "code",
   "id": "3d4c35905609659e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-06-18T20:35:14.813688Z",
     "start_time": "2025-06-18T20:35:14.797189Z"
    }
   },
   "source": [
    "samples_sub = samples[samples[\"sample-subject-subject-id\"].apply(lambda x: x in sub_ids)]\n",
    "samples_sub"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "             sample-id sample-tumor-type                sample-specimen  \\\n",
       "69    tcga-e2-a15i-01a            breast  :sample.specimen/fresh-frozen   \n",
       "70    tcga-e2-a15i-10a               NaN  :sample.specimen/fresh-frozen   \n",
       "71    tcga-e2-a15i-11a               NaN  :sample.specimen/fresh-frozen   \n",
       "105   tcga-b6-a1kn-01a            breast  :sample.specimen/fresh-frozen   \n",
       "106   tcga-b6-a1kn-10a               NaN  :sample.specimen/fresh-frozen   \n",
       "...                ...               ...                            ...   \n",
       "2135  tcga-ar-a1aq-01a            breast  :sample.specimen/fresh-frozen   \n",
       "2136  tcga-ar-a1aq-10a               NaN  :sample.specimen/fresh-frozen   \n",
       "2256  tcga-d8-a27e-10a               NaN  :sample.specimen/fresh-frozen   \n",
       "2284  tcga-e9-a1n9-11a               NaN  :sample.specimen/fresh-frozen   \n",
       "2293  tcga-ol-a5da-10a               NaN  :sample.specimen/fresh-frozen   \n",
       "\n",
       "     sample-metastasis                                     sample-uid  \\\n",
       "69               False  [tcga-brca, 6418023f5ea1c0660b0ae8e57363a1d8]   \n",
       "70               False  [tcga-brca, 3e3a7b86768f0bff5fabc547366d9091]   \n",
       "71               False  [tcga-brca, 9538fcf5d7e50fa6e8304367b61d704a]   \n",
       "105              False  [tcga-brca, 83fe2368405c6e3d2a08288d83f8f0db]   \n",
       "106              False  [tcga-brca, 870c4800388cc94b29d58203db3a065b]   \n",
       "...                ...                                            ...   \n",
       "2135             False  [tcga-brca, ba15ce01b2d7006adcf93c4c164fd668]   \n",
       "2136             False  [tcga-brca, 55cfb9a0147ff218289d4a58f76624a9]   \n",
       "2256             False  [tcga-brca, 3c207e60055076c99b42219f1cc01442]   \n",
       "2284             False  [tcga-brca, 93a83f1dd48b8968e49cd72d2eedac35]   \n",
       "2293             False  [tcga-brca, 7882611517385fe91546199a73b523c7]   \n",
       "\n",
       "               db-id sample-timepoint-timepoint-id sample-subject-subject-id  \\\n",
       "69    17592187539851            sample procurement              TCGA-E2-A15I   \n",
       "70    17592187540016            sample procurement              TCGA-E2-A15I   \n",
       "71    17592187540116            sample procurement              TCGA-E2-A15I   \n",
       "105   17592187547478            sample procurement              TCGA-B6-A1KN   \n",
       "106   17592187548097            sample procurement              TCGA-B6-A1KN   \n",
       "...              ...                           ...                       ...   \n",
       "2135  17592188092973            sample procurement              TCGA-AR-A1AQ   \n",
       "2136  17592188093360            sample procurement              TCGA-AR-A1AQ   \n",
       "2256  17592188109129            sample procurement              TCGA-D8-A27E   \n",
       "2284  17592188110096            sample procurement              TCGA-E9-A1N9   \n",
       "2293  17592188111439            sample procurement              TCGA-OL-A5DA   \n",
       "\n",
       "     sample-study-day-study-day-id  sample-type-db-id  \\\n",
       "69                               0     17592186045420   \n",
       "70                               0     17592186045421   \n",
       "71                               0     17592186045421   \n",
       "105                             36     17592186045420   \n",
       "106                             36     17592186045421   \n",
       "...                            ...                ...   \n",
       "2135                             8     17592186045420   \n",
       "2136                             8     17592186045421   \n",
       "2256                            41     17592186045421   \n",
       "2284                             0     17592186045421   \n",
       "2293                            33     17592186045421   \n",
       "\n",
       "     sample-gdc-anatomic-site-gdc-anatomic-site-name  \n",
       "69                                            Breast  \n",
       "70                                            Breast  \n",
       "71                                            Breast  \n",
       "105                                           Breast  \n",
       "106                                           Breast  \n",
       "...                                              ...  \n",
       "2135                                          Breast  \n",
       "2136                                          Breast  \n",
       "2256                                          Breast  \n",
       "2284                                          Breast  \n",
       "2293                                          Breast  \n",
       "\n",
       "[173 rows x 11 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>sample-tumor-type</th>\n",
       "      <th>sample-specimen</th>\n",
       "      <th>sample-metastasis</th>\n",
       "      <th>sample-uid</th>\n",
       "      <th>db-id</th>\n",
       "      <th>sample-timepoint-timepoint-id</th>\n",
       "      <th>sample-subject-subject-id</th>\n",
       "      <th>sample-study-day-study-day-id</th>\n",
       "      <th>sample-type-db-id</th>\n",
       "      <th>sample-gdc-anatomic-site-gdc-anatomic-site-name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>tcga-e2-a15i-01a</td>\n",
       "      <td>breast</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 6418023f5ea1c0660b0ae8e57363a1d8]</td>\n",
       "      <td>17592187539851</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-E2-A15I</td>\n",
       "      <td>0</td>\n",
       "      <td>17592186045420</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>tcga-e2-a15i-10a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 3e3a7b86768f0bff5fabc547366d9091]</td>\n",
       "      <td>17592187540016</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-E2-A15I</td>\n",
       "      <td>0</td>\n",
       "      <td>17592186045421</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>tcga-e2-a15i-11a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 9538fcf5d7e50fa6e8304367b61d704a]</td>\n",
       "      <td>17592187540116</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-E2-A15I</td>\n",
       "      <td>0</td>\n",
       "      <td>17592186045421</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>tcga-b6-a1kn-01a</td>\n",
       "      <td>breast</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 83fe2368405c6e3d2a08288d83f8f0db]</td>\n",
       "      <td>17592187547478</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-B6-A1KN</td>\n",
       "      <td>36</td>\n",
       "      <td>17592186045420</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>tcga-b6-a1kn-10a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 870c4800388cc94b29d58203db3a065b]</td>\n",
       "      <td>17592187548097</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-B6-A1KN</td>\n",
       "      <td>36</td>\n",
       "      <td>17592186045421</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2135</th>\n",
       "      <td>tcga-ar-a1aq-01a</td>\n",
       "      <td>breast</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, ba15ce01b2d7006adcf93c4c164fd668]</td>\n",
       "      <td>17592188092973</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-AR-A1AQ</td>\n",
       "      <td>8</td>\n",
       "      <td>17592186045420</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2136</th>\n",
       "      <td>tcga-ar-a1aq-10a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 55cfb9a0147ff218289d4a58f76624a9]</td>\n",
       "      <td>17592188093360</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-AR-A1AQ</td>\n",
       "      <td>8</td>\n",
       "      <td>17592186045421</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2256</th>\n",
       "      <td>tcga-d8-a27e-10a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 3c207e60055076c99b42219f1cc01442]</td>\n",
       "      <td>17592188109129</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-D8-A27E</td>\n",
       "      <td>41</td>\n",
       "      <td>17592186045421</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2284</th>\n",
       "      <td>tcga-e9-a1n9-11a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 93a83f1dd48b8968e49cd72d2eedac35]</td>\n",
       "      <td>17592188110096</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-E9-A1N9</td>\n",
       "      <td>0</td>\n",
       "      <td>17592186045421</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2293</th>\n",
       "      <td>tcga-ol-a5da-10a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>:sample.specimen/fresh-frozen</td>\n",
       "      <td>False</td>\n",
       "      <td>[tcga-brca, 7882611517385fe91546199a73b523c7]</td>\n",
       "      <td>17592188111439</td>\n",
       "      <td>sample procurement</td>\n",
       "      <td>TCGA-OL-A5DA</td>\n",
       "      <td>33</td>\n",
       "      <td>17592186045421</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>173 rows × 11 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 9
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": [
    "#### Clinical Data\n",
    "\n",
    "Because this is open access TCGA data, we only ahve access to the PFS, OS, and recurrence clinical data."
   ],
   "id": "1ab0cdb56c5fd0cc"
  },
  {
   "cell_type": "code",
   "id": "a2653774454e7b68",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-06-18T20:35:15.357250Z",
     "start_time": "2025-06-18T20:35:14.890879Z"
    }
   },
   "source": [
    "clin_sum = pqd.clinical_summary(\"tcga-brca\", db_name='tcga-brca')\n",
    "clin_sum"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "  clinical-observation-set-name\n",
       "0             pfs os recurrence"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>clinical-observation-set-name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>pfs os recurrence</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 10
  },
  {
   "cell_type": "code",
   "id": "44f9e8e55a7f7e88",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-06-18T20:35:15.414836Z",
     "start_time": "2025-06-18T20:35:15.402078Z"
    }
   },
   "source": [
    "samples_for_meas = assays_for_sub[assays_for_sub['measurement-set-name'] == 'baseline mutations']\n",
    "samples_for_meas"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "       subject-id         sample-id             assay-tech assay-name  \\\n",
       "4    TCGA-A8-A099  tcga-a8-a099-01a  :assay.technology/WES        WES   \n",
       "28   TCGA-A7-A0DA  tcga-a7-a0da-01a  :assay.technology/WES        WES   \n",
       "51   TCGA-AO-A12G  tcga-ao-a12g-01a  :assay.technology/WES        WES   \n",
       "62   TCGA-A2-A0EO  tcga-a2-a0eo-01a  :assay.technology/WES        WES   \n",
       "71   TCGA-A2-A04X  tcga-a2-a04x-01a  :assay.technology/WES        WES   \n",
       "..            ...               ...                    ...        ...   \n",
       "586  TCGA-E9-A1R4  tcga-e9-a1r4-01a  :assay.technology/WES        WES   \n",
       "588  TCGA-A8-A07B  tcga-a8-a07b-01a  :assay.technology/WES        WES   \n",
       "591  TCGA-A8-A06T  tcga-a8-a06t-01a  :assay.technology/WES        WES   \n",
       "597  TCGA-AC-A3W6  tcga-ac-a3w6-01a  :assay.technology/WES        WES   \n",
       "600  TCGA-AO-A0JD  tcga-ao-a0jd-01a  :assay.technology/WES        WES   \n",
       "\n",
       "    measurement-set-name  \n",
       "4     baseline mutations  \n",
       "28    baseline mutations  \n",
       "51    baseline mutations  \n",
       "62    baseline mutations  \n",
       "71    baseline mutations  \n",
       "..                   ...  \n",
       "586   baseline mutations  \n",
       "588   baseline mutations  \n",
       "591   baseline mutations  \n",
       "597   baseline mutations  \n",
       "600   baseline mutations  \n",
       "\n",
       "[82 rows x 5 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject-id</th>\n",
       "      <th>sample-id</th>\n",
       "      <th>assay-tech</th>\n",
       "      <th>assay-name</th>\n",
       "      <th>measurement-set-name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-A8-A099</td>\n",
       "      <td>tcga-a8-a099-01a</td>\n",
       "      <td>:assay.technology/WES</td>\n",
       "      <td>WES</td>\n",
       "      <td>baseline mutations</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>TCGA-A7-A0DA</td>\n",
       "      <td>tcga-a7-a0da-01a</td>\n",
       "      <td>:assay.technology/WES</td>\n",
       "      <td>WES</td>\n",
       "      <td>baseline mutations</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>TCGA-AO-A12G</td>\n",
       "      <td>tcga-ao-a12g-01a</td>\n",
       "      <td>:assay.technology/WES</td>\n",
       "      <td>WES</td>\n",
       "      <td>baseline mutations</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>TCGA-A2-A0EO</td>\n",
       "      <td>tcga-a2-a0eo-01a</td>\n",
       "      <td>:assay.technology/WES</td>\n",
       "      <td>WES</td>\n",
       "      <td>baseline mutations</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>TCGA-A2-A04X</td>\n",
       "      <td>tcga-a2-a04x-01a</td>\n",
       "      <td>:assay.technology/WES</td>\n",
       "      <td>WES</td>\n",
       "      <td>baseline mutations</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>586</th>\n",
       "      <td>TCGA-E9-A1R4</td>\n",
       "      <td>tcga-e9-a1r4-01a</td>\n",
       "      <td>:assay.technology/WES</td>\n",
       "      <td>WES</td>\n",
       "      <td>baseline mutations</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>588</th>\n",
       "      <td>TCGA-A8-A07B</td>\n",
       "      <td>tcga-a8-a07b-01a</td>\n",
       "      <td>:assay.technology/WES</td>\n",
       "      <td>WES</td>\n",
       "      <td>baseline mutations</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>591</th>\n",
       "      <td>TCGA-A8-A06T</td>\n",
       "      <td>tcga-a8-a06t-01a</td>\n",
       "      <td>:assay.technology/WES</td>\n",
       "      <td>WES</td>\n",
       "      <td>baseline mutations</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>597</th>\n",
       "      <td>TCGA-AC-A3W6</td>\n",
       "      <td>tcga-ac-a3w6-01a</td>\n",
       "      <td>:assay.technology/WES</td>\n",
       "      <td>WES</td>\n",
       "      <td>baseline mutations</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>600</th>\n",
       "      <td>TCGA-AO-A0JD</td>\n",
       "      <td>tcga-ao-a0jd-01a</td>\n",
       "      <td>:assay.technology/WES</td>\n",
       "      <td>WES</td>\n",
       "      <td>baseline mutations</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>82 rows × 5 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 11
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": [
    "#### Samples and Measurements\n",
    "\n",
    "This query selects the measurements for all samples that were included in a particular measurement set. In this case, the WES\n",
    "(whole exome sequencing) data at baseline, with measurement set name 'baseline mutations')."
   ],
   "id": "93b5cfb0662b985b"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-06-18T20:35:16.403708Z",
     "start_time": "2025-06-18T20:35:15.507751Z"
    }
   },
   "cell_type": "code",
   "source": [
    "meas_samples = pqd.sample_measurements(\"tcga-brca\", \"baseline mutations\",\n",
    "                                       samples_for_meas['sample-id'].tolist(), db_name='tcga-brca',\n",
    "                                       timeout=120)\n",
    "meas_samples"
   ],
   "id": "537c41c7e30d10ff",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "               db-id                                     measurement-id  \\\n",
       "0     17592234968693  tcga-a1-a0sh-01a-GRCh37:chr23:+:108911440:1089...   \n",
       "1     17592234968695  tcga-a1-a0sh-01a-GRCh37:chr1:+:247040513:24704...   \n",
       "2     17592234968697  tcga-a1-a0sh-01a-GRCh37:chr6:+:85400295:854002...   \n",
       "3     17592234968699  tcga-a1-a0sh-01a-GRCh37:chr2:+:61965629:619656...   \n",
       "4     17592234968701  tcga-a1-a0sh-01a-GRCh37:chr19:+:117865025:1178...   \n",
       "...              ...                                                ...   \n",
       "8028  17592235128883  tcga-ol-a66h-01a-GRCh37:chr3:+:12447485:124474...   \n",
       "8029  17592235128885  tcga-ol-a66h-01a-GRCh37:chr8:+:10464643:104646...   \n",
       "8030  17592235128887  tcga-ol-a66h-01a-GRCh37:chr17:+:19584880:19584...   \n",
       "8031  17592235128889  tcga-ol-a66h-01a-GRCh37:chr4:+:88411538:884115...   \n",
       "8032  17592235128891  tcga-ol-a66h-01a-GRCh37:chr10:+:63170376:63170...   \n",
       "\n",
       "                                    measurement-uid  \\\n",
       "0     [tcga-brca, a98f5fd31e7eb25a11aabcc29df6ebd7]   \n",
       "1     [tcga-brca, 04a0aebe4f1e1662f62cee204cbc74b8]   \n",
       "2     [tcga-brca, 2743c86cf741a47ec9aa882d755f01d8]   \n",
       "3     [tcga-brca, edbd0160cf703fb987f0810ebcb98d62]   \n",
       "4     [tcga-brca, cf54bb287c1d40d26e2b4af7551038de]   \n",
       "...                                             ...   \n",
       "8028  [tcga-brca, 62f30b4472c51f6dedf626fa89c3cf34]   \n",
       "8029  [tcga-brca, d584100cdf5d68946a6a47e3254d550e]   \n",
       "8030  [tcga-brca, 23aa10c05ed127c8ea6ce17a3b3a6f29]   \n",
       "8031  [tcga-brca, 985874c25fcacc499fb1b59e6eecf464]   \n",
       "8032  [tcga-brca, 46d6155733290f7acea12f7a190abec7]   \n",
       "\n",
       "     measurement-sample-sample-id          measurement-variant-variant-id  \n",
       "0                tcga-a1-a0sh-01a  GRCh37:chr23:+:108911440:108911440/C/T  \n",
       "1                tcga-a1-a0sh-01a   GRCh37:chr1:+:247040513:247040513/C/A  \n",
       "2                tcga-a1-a0sh-01a     GRCh37:chr6:+:85400295:85400295/G/A  \n",
       "3                tcga-a1-a0sh-01a     GRCh37:chr2:+:61965629:61965629/T/C  \n",
       "4                tcga-a1-a0sh-01a  GRCh37:chr19:+:117865025:117865025/C/G  \n",
       "...                           ...                                     ...  \n",
       "8028             tcga-ol-a66h-01a     GRCh37:chr3:+:12447485:12447485/C/G  \n",
       "8029             tcga-ol-a66h-01a     GRCh37:chr8:+:10464643:10464643/T/A  \n",
       "8030             tcga-ol-a66h-01a    GRCh37:chr17:+:19584880:19584880/A/G  \n",
       "8031             tcga-ol-a66h-01a     GRCh37:chr4:+:88411538:88411538/C/G  \n",
       "8032             tcga-ol-a66h-01a    GRCh37:chr10:+:63170376:63170376/C/G  \n",
       "\n",
       "[8033 rows x 5 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>db-id</th>\n",
       "      <th>measurement-id</th>\n",
       "      <th>measurement-uid</th>\n",
       "      <th>measurement-sample-sample-id</th>\n",
       "      <th>measurement-variant-variant-id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17592234968693</td>\n",
       "      <td>tcga-a1-a0sh-01a-GRCh37:chr23:+:108911440:1089...</td>\n",
       "      <td>[tcga-brca, a98f5fd31e7eb25a11aabcc29df6ebd7]</td>\n",
       "      <td>tcga-a1-a0sh-01a</td>\n",
       "      <td>GRCh37:chr23:+:108911440:108911440/C/T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>17592234968695</td>\n",
       "      <td>tcga-a1-a0sh-01a-GRCh37:chr1:+:247040513:24704...</td>\n",
       "      <td>[tcga-brca, 04a0aebe4f1e1662f62cee204cbc74b8]</td>\n",
       "      <td>tcga-a1-a0sh-01a</td>\n",
       "      <td>GRCh37:chr1:+:247040513:247040513/C/A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>17592234968697</td>\n",
       "      <td>tcga-a1-a0sh-01a-GRCh37:chr6:+:85400295:854002...</td>\n",
       "      <td>[tcga-brca, 2743c86cf741a47ec9aa882d755f01d8]</td>\n",
       "      <td>tcga-a1-a0sh-01a</td>\n",
       "      <td>GRCh37:chr6:+:85400295:85400295/G/A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>17592234968699</td>\n",
       "      <td>tcga-a1-a0sh-01a-GRCh37:chr2:+:61965629:619656...</td>\n",
       "      <td>[tcga-brca, edbd0160cf703fb987f0810ebcb98d62]</td>\n",
       "      <td>tcga-a1-a0sh-01a</td>\n",
       "      <td>GRCh37:chr2:+:61965629:61965629/T/C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>17592234968701</td>\n",
       "      <td>tcga-a1-a0sh-01a-GRCh37:chr19:+:117865025:1178...</td>\n",
       "      <td>[tcga-brca, cf54bb287c1d40d26e2b4af7551038de]</td>\n",
       "      <td>tcga-a1-a0sh-01a</td>\n",
       "      <td>GRCh37:chr19:+:117865025:117865025/C/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8028</th>\n",
       "      <td>17592235128883</td>\n",
       "      <td>tcga-ol-a66h-01a-GRCh37:chr3:+:12447485:124474...</td>\n",
       "      <td>[tcga-brca, 62f30b4472c51f6dedf626fa89c3cf34]</td>\n",
       "      <td>tcga-ol-a66h-01a</td>\n",
       "      <td>GRCh37:chr3:+:12447485:12447485/C/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8029</th>\n",
       "      <td>17592235128885</td>\n",
       "      <td>tcga-ol-a66h-01a-GRCh37:chr8:+:10464643:104646...</td>\n",
       "      <td>[tcga-brca, d584100cdf5d68946a6a47e3254d550e]</td>\n",
       "      <td>tcga-ol-a66h-01a</td>\n",
       "      <td>GRCh37:chr8:+:10464643:10464643/T/A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8030</th>\n",
       "      <td>17592235128887</td>\n",
       "      <td>tcga-ol-a66h-01a-GRCh37:chr17:+:19584880:19584...</td>\n",
       "      <td>[tcga-brca, 23aa10c05ed127c8ea6ce17a3b3a6f29]</td>\n",
       "      <td>tcga-ol-a66h-01a</td>\n",
       "      <td>GRCh37:chr17:+:19584880:19584880/A/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8031</th>\n",
       "      <td>17592235128889</td>\n",
       "      <td>tcga-ol-a66h-01a-GRCh37:chr4:+:88411538:884115...</td>\n",
       "      <td>[tcga-brca, 985874c25fcacc499fb1b59e6eecf464]</td>\n",
       "      <td>tcga-ol-a66h-01a</td>\n",
       "      <td>GRCh37:chr4:+:88411538:88411538/C/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8032</th>\n",
       "      <td>17592235128891</td>\n",
       "      <td>tcga-ol-a66h-01a-GRCh37:chr10:+:63170376:63170...</td>\n",
       "      <td>[tcga-brca, 46d6155733290f7acea12f7a190abec7]</td>\n",
       "      <td>tcga-ol-a66h-01a</td>\n",
       "      <td>GRCh37:chr10:+:63170376:63170376/C/G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8033 rows × 5 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 12
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": [
    "#### Variant Data\n",
    "\n",
    "Here we access variant data for each of the samples "
   ],
   "id": "7c7ed792e33e493a"
  },
  {
   "cell_type": "code",
   "id": "3013e87d1d690659",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-06-18T20:35:17.860510Z",
     "start_time": "2025-06-18T20:35:16.441488Z"
    }
   },
   "source": [
    "var = pqr.variants(meas_samples['measurement-variant-variant-id'].tolist(), db_name='tcga-brca')\n",
    "var"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           variant-type variant-ref-allele  \\\n",
       "0     :variant.type/snp                  C   \n",
       "1     :variant.type/snp                  C   \n",
       "2     :variant.type/snp                  G   \n",
       "3     :variant.type/snp                  T   \n",
       "4     :variant.type/snp                  C   \n",
       "...                 ...                ...   \n",
       "8014  :variant.type/snp                  C   \n",
       "8015  :variant.type/snp                  T   \n",
       "8016  :variant.type/snp                  A   \n",
       "8017  :variant.type/snp                  C   \n",
       "8018  :variant.type/snp                  C   \n",
       "\n",
       "                                  variant-id variant-alt-allele  \\\n",
       "0     GRCh37:chr23:+:108911440:108911440/C/T                  T   \n",
       "1      GRCh37:chr1:+:247040513:247040513/C/A                  A   \n",
       "2        GRCh37:chr6:+:85400295:85400295/G/A                  A   \n",
       "3        GRCh37:chr2:+:61965629:61965629/T/C                  C   \n",
       "4     GRCh37:chr19:+:117865025:117865025/C/G                  G   \n",
       "...                                      ...                ...   \n",
       "8014     GRCh37:chr3:+:12447485:12447485/C/G                  G   \n",
       "8015     GRCh37:chr8:+:10464643:10464643/T/A                  A   \n",
       "8016    GRCh37:chr17:+:19584880:19584880/A/G                  G   \n",
       "8017     GRCh37:chr4:+:88411538:88411538/C/G                  G   \n",
       "8018    GRCh37:chr10:+:63170376:63170376/C/G                  G   \n",
       "\n",
       "               db-id            variant-classification  \\\n",
       "0     17592187348366  :variant.classification/missense   \n",
       "1     17592187348368  :variant.classification/nonsense   \n",
       "2     17592187348370  :variant.classification/missense   \n",
       "3     17592187348372  :variant.classification/missense   \n",
       "4     17592187348374  :variant.classification/missense   \n",
       "...              ...                               ...   \n",
       "8014  17592187508556  :variant.classification/missense   \n",
       "8015  17592187508558  :variant.classification/missense   \n",
       "8016  17592187508560  :variant.classification/missense   \n",
       "8017  17592187508562  :variant.classification/missense   \n",
       "8018  17592187508564  :variant.classification/missense   \n",
       "\n",
       "     variant-gene-gene-hgnc-symbol  \\\n",
       "0                            CNTN4   \n",
       "1                          COL14A1   \n",
       "2                           COL7A1   \n",
       "3                            CPEB2   \n",
       "4                            CSMD1   \n",
       "...                            ...   \n",
       "8014                         PARP8   \n",
       "8015                          ASB9   \n",
       "8016                        ZNF660   \n",
       "8017                       SIPA1L2   \n",
       "8018                         PRR33   \n",
       "\n",
       "     variant-genomic-coordinates-genomic-coordinate-id  \\\n",
       "0                   GRCh37:chr23:+:108911440:108911440   \n",
       "1                    GRCh37:chr1:+:247040513:247040513   \n",
       "2                      GRCh37:chr6:+:85400295:85400295   \n",
       "3                      GRCh37:chr2:+:61965629:61965629   \n",
       "4                   GRCh37:chr19:+:117865025:117865025   \n",
       "...                                                ...   \n",
       "8014                   GRCh37:chr3:+:12447485:12447485   \n",
       "8015                   GRCh37:chr8:+:10464643:10464643   \n",
       "8016                  GRCh37:chr17:+:19584880:19584880   \n",
       "8017                   GRCh37:chr4:+:88411538:88411538   \n",
       "8018                  GRCh37:chr10:+:63170376:63170376   \n",
       "\n",
       "      unify-import-most-recent-db-id  \n",
       "0                     17592186177812  \n",
       "1                     17592186177812  \n",
       "2                     17592186177812  \n",
       "3                     17592186177812  \n",
       "4                     17592186177812  \n",
       "...                              ...  \n",
       "8014                  17592186177812  \n",
       "8015                  17592186177812  \n",
       "8016                  17592186177812  \n",
       "8017                  17592186177812  \n",
       "8018                  17592186177812  \n",
       "\n",
       "[8019 rows x 9 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant-type</th>\n",
       "      <th>variant-ref-allele</th>\n",
       "      <th>variant-id</th>\n",
       "      <th>variant-alt-allele</th>\n",
       "      <th>db-id</th>\n",
       "      <th>variant-classification</th>\n",
       "      <th>variant-gene-gene-hgnc-symbol</th>\n",
       "      <th>variant-genomic-coordinates-genomic-coordinate-id</th>\n",
       "      <th>unify-import-most-recent-db-id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr23:+:108911440:108911440/C/T</td>\n",
       "      <td>T</td>\n",
       "      <td>17592187348366</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>CNTN4</td>\n",
       "      <td>GRCh37:chr23:+:108911440:108911440</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr1:+:247040513:247040513/C/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187348368</td>\n",
       "      <td>:variant.classification/nonsense</td>\n",
       "      <td>COL14A1</td>\n",
       "      <td>GRCh37:chr1:+:247040513:247040513</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>G</td>\n",
       "      <td>GRCh37:chr6:+:85400295:85400295/G/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187348370</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>COL7A1</td>\n",
       "      <td>GRCh37:chr6:+:85400295:85400295</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>T</td>\n",
       "      <td>GRCh37:chr2:+:61965629:61965629/T/C</td>\n",
       "      <td>C</td>\n",
       "      <td>17592187348372</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>CPEB2</td>\n",
       "      <td>GRCh37:chr2:+:61965629:61965629</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr19:+:117865025:117865025/C/G</td>\n",
       "      <td>G</td>\n",
       "      <td>17592187348374</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>CSMD1</td>\n",
       "      <td>GRCh37:chr19:+:117865025:117865025</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8014</th>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr3:+:12447485:12447485/C/G</td>\n",
       "      <td>G</td>\n",
       "      <td>17592187508556</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>PARP8</td>\n",
       "      <td>GRCh37:chr3:+:12447485:12447485</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8015</th>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>T</td>\n",
       "      <td>GRCh37:chr8:+:10464643:10464643/T/A</td>\n",
       "      <td>A</td>\n",
       "      <td>17592187508558</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>ASB9</td>\n",
       "      <td>GRCh37:chr8:+:10464643:10464643</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8016</th>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>A</td>\n",
       "      <td>GRCh37:chr17:+:19584880:19584880/A/G</td>\n",
       "      <td>G</td>\n",
       "      <td>17592187508560</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>ZNF660</td>\n",
       "      <td>GRCh37:chr17:+:19584880:19584880</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8017</th>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr4:+:88411538:88411538/C/G</td>\n",
       "      <td>G</td>\n",
       "      <td>17592187508562</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>SIPA1L2</td>\n",
       "      <td>GRCh37:chr4:+:88411538:88411538</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8018</th>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>C</td>\n",
       "      <td>GRCh37:chr10:+:63170376:63170376/C/G</td>\n",
       "      <td>G</td>\n",
       "      <td>17592187508564</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "      <td>PRR33</td>\n",
       "      <td>GRCh37:chr10:+:63170376:63170376</td>\n",
       "      <td>17592186177812</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8019 rows × 9 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 13
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": [
    "As is often the case with the query wrappers, we now have two tables (sets of tuples) that need to be\n",
    "joined together with the `merge` method in `pandas.DataFrame`, so that all measurement, variant, and\n",
    "sample data is linked."
   ],
   "id": "ca41aa1b4901e64d"
  },
  {
   "cell_type": "code",
   "id": "9c34888ffd0dd9ca",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-06-18T20:35:17.916425Z",
     "start_time": "2025-06-18T20:35:17.900074Z"
    }
   },
   "source": [
    "w_var = meas_samples.merge(\n",
    "    var[['variant-id',\n",
    "         'variant-ref-allele',\n",
    "         'variant-alt-allele',\n",
    "         'variant-gene-gene-hgnc-symbol',\n",
    "         'variant-genomic-coordinates-genomic-coordinate-id',\n",
    "         'variant-type',\n",
    "         'variant-classification']],\n",
    "    left_on='measurement-variant-variant-id',\n",
    "    right_on='variant-id',\n",
    "    how='left'\n",
    ")\n",
    "w_var"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "               db-id                                     measurement-id  \\\n",
       "0     17592234968693  tcga-a1-a0sh-01a-GRCh37:chr23:+:108911440:1089...   \n",
       "1     17592234968695  tcga-a1-a0sh-01a-GRCh37:chr1:+:247040513:24704...   \n",
       "2     17592234968697  tcga-a1-a0sh-01a-GRCh37:chr6:+:85400295:854002...   \n",
       "3     17592234968699  tcga-a1-a0sh-01a-GRCh37:chr2:+:61965629:619656...   \n",
       "4     17592234968701  tcga-a1-a0sh-01a-GRCh37:chr19:+:117865025:1178...   \n",
       "...              ...                                                ...   \n",
       "8028  17592235128883  tcga-ol-a66h-01a-GRCh37:chr3:+:12447485:124474...   \n",
       "8029  17592235128885  tcga-ol-a66h-01a-GRCh37:chr8:+:10464643:104646...   \n",
       "8030  17592235128887  tcga-ol-a66h-01a-GRCh37:chr17:+:19584880:19584...   \n",
       "8031  17592235128889  tcga-ol-a66h-01a-GRCh37:chr4:+:88411538:884115...   \n",
       "8032  17592235128891  tcga-ol-a66h-01a-GRCh37:chr10:+:63170376:63170...   \n",
       "\n",
       "                                    measurement-uid  \\\n",
       "0     [tcga-brca, a98f5fd31e7eb25a11aabcc29df6ebd7]   \n",
       "1     [tcga-brca, 04a0aebe4f1e1662f62cee204cbc74b8]   \n",
       "2     [tcga-brca, 2743c86cf741a47ec9aa882d755f01d8]   \n",
       "3     [tcga-brca, edbd0160cf703fb987f0810ebcb98d62]   \n",
       "4     [tcga-brca, cf54bb287c1d40d26e2b4af7551038de]   \n",
       "...                                             ...   \n",
       "8028  [tcga-brca, 62f30b4472c51f6dedf626fa89c3cf34]   \n",
       "8029  [tcga-brca, d584100cdf5d68946a6a47e3254d550e]   \n",
       "8030  [tcga-brca, 23aa10c05ed127c8ea6ce17a3b3a6f29]   \n",
       "8031  [tcga-brca, 985874c25fcacc499fb1b59e6eecf464]   \n",
       "8032  [tcga-brca, 46d6155733290f7acea12f7a190abec7]   \n",
       "\n",
       "     measurement-sample-sample-id          measurement-variant-variant-id  \\\n",
       "0                tcga-a1-a0sh-01a  GRCh37:chr23:+:108911440:108911440/C/T   \n",
       "1                tcga-a1-a0sh-01a   GRCh37:chr1:+:247040513:247040513/C/A   \n",
       "2                tcga-a1-a0sh-01a     GRCh37:chr6:+:85400295:85400295/G/A   \n",
       "3                tcga-a1-a0sh-01a     GRCh37:chr2:+:61965629:61965629/T/C   \n",
       "4                tcga-a1-a0sh-01a  GRCh37:chr19:+:117865025:117865025/C/G   \n",
       "...                           ...                                     ...   \n",
       "8028             tcga-ol-a66h-01a     GRCh37:chr3:+:12447485:12447485/C/G   \n",
       "8029             tcga-ol-a66h-01a     GRCh37:chr8:+:10464643:10464643/T/A   \n",
       "8030             tcga-ol-a66h-01a    GRCh37:chr17:+:19584880:19584880/A/G   \n",
       "8031             tcga-ol-a66h-01a     GRCh37:chr4:+:88411538:88411538/C/G   \n",
       "8032             tcga-ol-a66h-01a    GRCh37:chr10:+:63170376:63170376/C/G   \n",
       "\n",
       "                                  variant-id variant-ref-allele  \\\n",
       "0     GRCh37:chr23:+:108911440:108911440/C/T                  C   \n",
       "1      GRCh37:chr1:+:247040513:247040513/C/A                  C   \n",
       "2        GRCh37:chr6:+:85400295:85400295/G/A                  G   \n",
       "3        GRCh37:chr2:+:61965629:61965629/T/C                  T   \n",
       "4     GRCh37:chr19:+:117865025:117865025/C/G                  C   \n",
       "...                                      ...                ...   \n",
       "8028     GRCh37:chr3:+:12447485:12447485/C/G                  C   \n",
       "8029     GRCh37:chr8:+:10464643:10464643/T/A                  T   \n",
       "8030    GRCh37:chr17:+:19584880:19584880/A/G                  A   \n",
       "8031     GRCh37:chr4:+:88411538:88411538/C/G                  C   \n",
       "8032    GRCh37:chr10:+:63170376:63170376/C/G                  C   \n",
       "\n",
       "     variant-alt-allele variant-gene-gene-hgnc-symbol  \\\n",
       "0                     T                         CNTN4   \n",
       "1                     A                       COL14A1   \n",
       "2                     A                        COL7A1   \n",
       "3                     C                         CPEB2   \n",
       "4                     G                         CSMD1   \n",
       "...                 ...                           ...   \n",
       "8028                  G                         PARP8   \n",
       "8029                  A                          ASB9   \n",
       "8030                  G                        ZNF660   \n",
       "8031                  G                       SIPA1L2   \n",
       "8032                  G                         PRR33   \n",
       "\n",
       "     variant-genomic-coordinates-genomic-coordinate-id       variant-type  \\\n",
       "0                   GRCh37:chr23:+:108911440:108911440  :variant.type/snp   \n",
       "1                    GRCh37:chr1:+:247040513:247040513  :variant.type/snp   \n",
       "2                      GRCh37:chr6:+:85400295:85400295  :variant.type/snp   \n",
       "3                      GRCh37:chr2:+:61965629:61965629  :variant.type/snp   \n",
       "4                   GRCh37:chr19:+:117865025:117865025  :variant.type/snp   \n",
       "...                                                ...                ...   \n",
       "8028                   GRCh37:chr3:+:12447485:12447485  :variant.type/snp   \n",
       "8029                   GRCh37:chr8:+:10464643:10464643  :variant.type/snp   \n",
       "8030                  GRCh37:chr17:+:19584880:19584880  :variant.type/snp   \n",
       "8031                   GRCh37:chr4:+:88411538:88411538  :variant.type/snp   \n",
       "8032                  GRCh37:chr10:+:63170376:63170376  :variant.type/snp   \n",
       "\n",
       "                variant-classification  \n",
       "0     :variant.classification/missense  \n",
       "1     :variant.classification/nonsense  \n",
       "2     :variant.classification/missense  \n",
       "3     :variant.classification/missense  \n",
       "4     :variant.classification/missense  \n",
       "...                                ...  \n",
       "8028  :variant.classification/missense  \n",
       "8029  :variant.classification/missense  \n",
       "8030  :variant.classification/missense  \n",
       "8031  :variant.classification/missense  \n",
       "8032  :variant.classification/missense  \n",
       "\n",
       "[8033 rows x 12 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>db-id</th>\n",
       "      <th>measurement-id</th>\n",
       "      <th>measurement-uid</th>\n",
       "      <th>measurement-sample-sample-id</th>\n",
       "      <th>measurement-variant-variant-id</th>\n",
       "      <th>variant-id</th>\n",
       "      <th>variant-ref-allele</th>\n",
       "      <th>variant-alt-allele</th>\n",
       "      <th>variant-gene-gene-hgnc-symbol</th>\n",
       "      <th>variant-genomic-coordinates-genomic-coordinate-id</th>\n",
       "      <th>variant-type</th>\n",
       "      <th>variant-classification</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17592234968693</td>\n",
       "      <td>tcga-a1-a0sh-01a-GRCh37:chr23:+:108911440:1089...</td>\n",
       "      <td>[tcga-brca, a98f5fd31e7eb25a11aabcc29df6ebd7]</td>\n",
       "      <td>tcga-a1-a0sh-01a</td>\n",
       "      <td>GRCh37:chr23:+:108911440:108911440/C/T</td>\n",
       "      <td>GRCh37:chr23:+:108911440:108911440/C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CNTN4</td>\n",
       "      <td>GRCh37:chr23:+:108911440:108911440</td>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>17592234968695</td>\n",
       "      <td>tcga-a1-a0sh-01a-GRCh37:chr1:+:247040513:24704...</td>\n",
       "      <td>[tcga-brca, 04a0aebe4f1e1662f62cee204cbc74b8]</td>\n",
       "      <td>tcga-a1-a0sh-01a</td>\n",
       "      <td>GRCh37:chr1:+:247040513:247040513/C/A</td>\n",
       "      <td>GRCh37:chr1:+:247040513:247040513/C/A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>COL14A1</td>\n",
       "      <td>GRCh37:chr1:+:247040513:247040513</td>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>:variant.classification/nonsense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>17592234968697</td>\n",
       "      <td>tcga-a1-a0sh-01a-GRCh37:chr6:+:85400295:854002...</td>\n",
       "      <td>[tcga-brca, 2743c86cf741a47ec9aa882d755f01d8]</td>\n",
       "      <td>tcga-a1-a0sh-01a</td>\n",
       "      <td>GRCh37:chr6:+:85400295:85400295/G/A</td>\n",
       "      <td>GRCh37:chr6:+:85400295:85400295/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>COL7A1</td>\n",
       "      <td>GRCh37:chr6:+:85400295:85400295</td>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>17592234968699</td>\n",
       "      <td>tcga-a1-a0sh-01a-GRCh37:chr2:+:61965629:619656...</td>\n",
       "      <td>[tcga-brca, edbd0160cf703fb987f0810ebcb98d62]</td>\n",
       "      <td>tcga-a1-a0sh-01a</td>\n",
       "      <td>GRCh37:chr2:+:61965629:61965629/T/C</td>\n",
       "      <td>GRCh37:chr2:+:61965629:61965629/T/C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>CPEB2</td>\n",
       "      <td>GRCh37:chr2:+:61965629:61965629</td>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>17592234968701</td>\n",
       "      <td>tcga-a1-a0sh-01a-GRCh37:chr19:+:117865025:1178...</td>\n",
       "      <td>[tcga-brca, cf54bb287c1d40d26e2b4af7551038de]</td>\n",
       "      <td>tcga-a1-a0sh-01a</td>\n",
       "      <td>GRCh37:chr19:+:117865025:117865025/C/G</td>\n",
       "      <td>GRCh37:chr19:+:117865025:117865025/C/G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>CSMD1</td>\n",
       "      <td>GRCh37:chr19:+:117865025:117865025</td>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8028</th>\n",
       "      <td>17592235128883</td>\n",
       "      <td>tcga-ol-a66h-01a-GRCh37:chr3:+:12447485:124474...</td>\n",
       "      <td>[tcga-brca, 62f30b4472c51f6dedf626fa89c3cf34]</td>\n",
       "      <td>tcga-ol-a66h-01a</td>\n",
       "      <td>GRCh37:chr3:+:12447485:12447485/C/G</td>\n",
       "      <td>GRCh37:chr3:+:12447485:12447485/C/G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PARP8</td>\n",
       "      <td>GRCh37:chr3:+:12447485:12447485</td>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8029</th>\n",
       "      <td>17592235128885</td>\n",
       "      <td>tcga-ol-a66h-01a-GRCh37:chr8:+:10464643:104646...</td>\n",
       "      <td>[tcga-brca, d584100cdf5d68946a6a47e3254d550e]</td>\n",
       "      <td>tcga-ol-a66h-01a</td>\n",
       "      <td>GRCh37:chr8:+:10464643:10464643/T/A</td>\n",
       "      <td>GRCh37:chr8:+:10464643:10464643/T/A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>ASB9</td>\n",
       "      <td>GRCh37:chr8:+:10464643:10464643</td>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8030</th>\n",
       "      <td>17592235128887</td>\n",
       "      <td>tcga-ol-a66h-01a-GRCh37:chr17:+:19584880:19584...</td>\n",
       "      <td>[tcga-brca, 23aa10c05ed127c8ea6ce17a3b3a6f29]</td>\n",
       "      <td>tcga-ol-a66h-01a</td>\n",
       "      <td>GRCh37:chr17:+:19584880:19584880/A/G</td>\n",
       "      <td>GRCh37:chr17:+:19584880:19584880/A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>ZNF660</td>\n",
       "      <td>GRCh37:chr17:+:19584880:19584880</td>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8031</th>\n",
       "      <td>17592235128889</td>\n",
       "      <td>tcga-ol-a66h-01a-GRCh37:chr4:+:88411538:884115...</td>\n",
       "      <td>[tcga-brca, 985874c25fcacc499fb1b59e6eecf464]</td>\n",
       "      <td>tcga-ol-a66h-01a</td>\n",
       "      <td>GRCh37:chr4:+:88411538:88411538/C/G</td>\n",
       "      <td>GRCh37:chr4:+:88411538:88411538/C/G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SIPA1L2</td>\n",
       "      <td>GRCh37:chr4:+:88411538:88411538</td>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8032</th>\n",
       "      <td>17592235128891</td>\n",
       "      <td>tcga-ol-a66h-01a-GRCh37:chr10:+:63170376:63170...</td>\n",
       "      <td>[tcga-brca, 46d6155733290f7acea12f7a190abec7]</td>\n",
       "      <td>tcga-ol-a66h-01a</td>\n",
       "      <td>GRCh37:chr10:+:63170376:63170376/C/G</td>\n",
       "      <td>GRCh37:chr10:+:63170376:63170376/C/G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PRR33</td>\n",
       "      <td>GRCh37:chr10:+:63170376:63170376</td>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8033 rows × 12 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 14
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "With that additional context, we can narrow the query down to look at missense mutations only.",
   "id": "d9a67be001870e78"
  },
  {
   "cell_type": "code",
   "id": "8f53087db8f6e263",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-06-18T20:35:17.972883Z",
     "start_time": "2025-06-18T20:35:17.958269Z"
    }
   },
   "source": [
    "w_var[w_var['variant-classification'] == ':variant.classification/missense']"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "               db-id                                     measurement-id  \\\n",
       "0     17592234968693  tcga-a1-a0sh-01a-GRCh37:chr23:+:108911440:1089...   \n",
       "2     17592234968697  tcga-a1-a0sh-01a-GRCh37:chr6:+:85400295:854002...   \n",
       "3     17592234968699  tcga-a1-a0sh-01a-GRCh37:chr2:+:61965629:619656...   \n",
       "4     17592234968701  tcga-a1-a0sh-01a-GRCh37:chr19:+:117865025:1178...   \n",
       "5     17592234968703  tcga-a1-a0sh-01a-GRCh37:chr7:+:28506628:285066...   \n",
       "...              ...                                                ...   \n",
       "8028  17592235128883  tcga-ol-a66h-01a-GRCh37:chr3:+:12447485:124474...   \n",
       "8029  17592235128885  tcga-ol-a66h-01a-GRCh37:chr8:+:10464643:104646...   \n",
       "8030  17592235128887  tcga-ol-a66h-01a-GRCh37:chr17:+:19584880:19584...   \n",
       "8031  17592235128889  tcga-ol-a66h-01a-GRCh37:chr4:+:88411538:884115...   \n",
       "8032  17592235128891  tcga-ol-a66h-01a-GRCh37:chr10:+:63170376:63170...   \n",
       "\n",
       "                                    measurement-uid  \\\n",
       "0     [tcga-brca, a98f5fd31e7eb25a11aabcc29df6ebd7]   \n",
       "2     [tcga-brca, 2743c86cf741a47ec9aa882d755f01d8]   \n",
       "3     [tcga-brca, edbd0160cf703fb987f0810ebcb98d62]   \n",
       "4     [tcga-brca, cf54bb287c1d40d26e2b4af7551038de]   \n",
       "5     [tcga-brca, 5d45e3e29c3ffaf60e4314c87ecb5241]   \n",
       "...                                             ...   \n",
       "8028  [tcga-brca, 62f30b4472c51f6dedf626fa89c3cf34]   \n",
       "8029  [tcga-brca, d584100cdf5d68946a6a47e3254d550e]   \n",
       "8030  [tcga-brca, 23aa10c05ed127c8ea6ce17a3b3a6f29]   \n",
       "8031  [tcga-brca, 985874c25fcacc499fb1b59e6eecf464]   \n",
       "8032  [tcga-brca, 46d6155733290f7acea12f7a190abec7]   \n",
       "\n",
       "     measurement-sample-sample-id          measurement-variant-variant-id  \\\n",
       "0                tcga-a1-a0sh-01a  GRCh37:chr23:+:108911440:108911440/C/T   \n",
       "2                tcga-a1-a0sh-01a     GRCh37:chr6:+:85400295:85400295/G/A   \n",
       "3                tcga-a1-a0sh-01a     GRCh37:chr2:+:61965629:61965629/T/C   \n",
       "4                tcga-a1-a0sh-01a  GRCh37:chr19:+:117865025:117865025/C/G   \n",
       "5                tcga-a1-a0sh-01a     GRCh37:chr7:+:28506628:28506628/G/T   \n",
       "...                           ...                                     ...   \n",
       "8028             tcga-ol-a66h-01a     GRCh37:chr3:+:12447485:12447485/C/G   \n",
       "8029             tcga-ol-a66h-01a     GRCh37:chr8:+:10464643:10464643/T/A   \n",
       "8030             tcga-ol-a66h-01a    GRCh37:chr17:+:19584880:19584880/A/G   \n",
       "8031             tcga-ol-a66h-01a     GRCh37:chr4:+:88411538:88411538/C/G   \n",
       "8032             tcga-ol-a66h-01a    GRCh37:chr10:+:63170376:63170376/C/G   \n",
       "\n",
       "                                  variant-id variant-ref-allele  \\\n",
       "0     GRCh37:chr23:+:108911440:108911440/C/T                  C   \n",
       "2        GRCh37:chr6:+:85400295:85400295/G/A                  G   \n",
       "3        GRCh37:chr2:+:61965629:61965629/T/C                  T   \n",
       "4     GRCh37:chr19:+:117865025:117865025/C/G                  C   \n",
       "5        GRCh37:chr7:+:28506628:28506628/G/T                  G   \n",
       "...                                      ...                ...   \n",
       "8028     GRCh37:chr3:+:12447485:12447485/C/G                  C   \n",
       "8029     GRCh37:chr8:+:10464643:10464643/T/A                  T   \n",
       "8030    GRCh37:chr17:+:19584880:19584880/A/G                  A   \n",
       "8031     GRCh37:chr4:+:88411538:88411538/C/G                  C   \n",
       "8032    GRCh37:chr10:+:63170376:63170376/C/G                  C   \n",
       "\n",
       "     variant-alt-allele variant-gene-gene-hgnc-symbol  \\\n",
       "0                     T                         CNTN4   \n",
       "2                     A                        COL7A1   \n",
       "3                     C                         CPEB2   \n",
       "4                     G                         CSMD1   \n",
       "5                     T                          CUBN   \n",
       "...                 ...                           ...   \n",
       "8028                  G                         PARP8   \n",
       "8029                  A                          ASB9   \n",
       "8030                  G                        ZNF660   \n",
       "8031                  G                       SIPA1L2   \n",
       "8032                  G                         PRR33   \n",
       "\n",
       "     variant-genomic-coordinates-genomic-coordinate-id       variant-type  \\\n",
       "0                   GRCh37:chr23:+:108911440:108911440  :variant.type/snp   \n",
       "2                      GRCh37:chr6:+:85400295:85400295  :variant.type/snp   \n",
       "3                      GRCh37:chr2:+:61965629:61965629  :variant.type/snp   \n",
       "4                   GRCh37:chr19:+:117865025:117865025  :variant.type/snp   \n",
       "5                      GRCh37:chr7:+:28506628:28506628  :variant.type/snp   \n",
       "...                                                ...                ...   \n",
       "8028                   GRCh37:chr3:+:12447485:12447485  :variant.type/snp   \n",
       "8029                   GRCh37:chr8:+:10464643:10464643  :variant.type/snp   \n",
       "8030                  GRCh37:chr17:+:19584880:19584880  :variant.type/snp   \n",
       "8031                   GRCh37:chr4:+:88411538:88411538  :variant.type/snp   \n",
       "8032                  GRCh37:chr10:+:63170376:63170376  :variant.type/snp   \n",
       "\n",
       "                variant-classification  \n",
       "0     :variant.classification/missense  \n",
       "2     :variant.classification/missense  \n",
       "3     :variant.classification/missense  \n",
       "4     :variant.classification/missense  \n",
       "5     :variant.classification/missense  \n",
       "...                                ...  \n",
       "8028  :variant.classification/missense  \n",
       "8029  :variant.classification/missense  \n",
       "8030  :variant.classification/missense  \n",
       "8031  :variant.classification/missense  \n",
       "8032  :variant.classification/missense  \n",
       "\n",
       "[5112 rows x 12 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>db-id</th>\n",
       "      <th>measurement-id</th>\n",
       "      <th>measurement-uid</th>\n",
       "      <th>measurement-sample-sample-id</th>\n",
       "      <th>measurement-variant-variant-id</th>\n",
       "      <th>variant-id</th>\n",
       "      <th>variant-ref-allele</th>\n",
       "      <th>variant-alt-allele</th>\n",
       "      <th>variant-gene-gene-hgnc-symbol</th>\n",
       "      <th>variant-genomic-coordinates-genomic-coordinate-id</th>\n",
       "      <th>variant-type</th>\n",
       "      <th>variant-classification</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17592234968693</td>\n",
       "      <td>tcga-a1-a0sh-01a-GRCh37:chr23:+:108911440:1089...</td>\n",
       "      <td>[tcga-brca, a98f5fd31e7eb25a11aabcc29df6ebd7]</td>\n",
       "      <td>tcga-a1-a0sh-01a</td>\n",
       "      <td>GRCh37:chr23:+:108911440:108911440/C/T</td>\n",
       "      <td>GRCh37:chr23:+:108911440:108911440/C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CNTN4</td>\n",
       "      <td>GRCh37:chr23:+:108911440:108911440</td>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>17592234968697</td>\n",
       "      <td>tcga-a1-a0sh-01a-GRCh37:chr6:+:85400295:854002...</td>\n",
       "      <td>[tcga-brca, 2743c86cf741a47ec9aa882d755f01d8]</td>\n",
       "      <td>tcga-a1-a0sh-01a</td>\n",
       "      <td>GRCh37:chr6:+:85400295:85400295/G/A</td>\n",
       "      <td>GRCh37:chr6:+:85400295:85400295/G/A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>COL7A1</td>\n",
       "      <td>GRCh37:chr6:+:85400295:85400295</td>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>17592234968699</td>\n",
       "      <td>tcga-a1-a0sh-01a-GRCh37:chr2:+:61965629:619656...</td>\n",
       "      <td>[tcga-brca, edbd0160cf703fb987f0810ebcb98d62]</td>\n",
       "      <td>tcga-a1-a0sh-01a</td>\n",
       "      <td>GRCh37:chr2:+:61965629:61965629/T/C</td>\n",
       "      <td>GRCh37:chr2:+:61965629:61965629/T/C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>CPEB2</td>\n",
       "      <td>GRCh37:chr2:+:61965629:61965629</td>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>17592234968701</td>\n",
       "      <td>tcga-a1-a0sh-01a-GRCh37:chr19:+:117865025:1178...</td>\n",
       "      <td>[tcga-brca, cf54bb287c1d40d26e2b4af7551038de]</td>\n",
       "      <td>tcga-a1-a0sh-01a</td>\n",
       "      <td>GRCh37:chr19:+:117865025:117865025/C/G</td>\n",
       "      <td>GRCh37:chr19:+:117865025:117865025/C/G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>CSMD1</td>\n",
       "      <td>GRCh37:chr19:+:117865025:117865025</td>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>17592234968703</td>\n",
       "      <td>tcga-a1-a0sh-01a-GRCh37:chr7:+:28506628:285066...</td>\n",
       "      <td>[tcga-brca, 5d45e3e29c3ffaf60e4314c87ecb5241]</td>\n",
       "      <td>tcga-a1-a0sh-01a</td>\n",
       "      <td>GRCh37:chr7:+:28506628:28506628/G/T</td>\n",
       "      <td>GRCh37:chr7:+:28506628:28506628/G/T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>CUBN</td>\n",
       "      <td>GRCh37:chr7:+:28506628:28506628</td>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8028</th>\n",
       "      <td>17592235128883</td>\n",
       "      <td>tcga-ol-a66h-01a-GRCh37:chr3:+:12447485:124474...</td>\n",
       "      <td>[tcga-brca, 62f30b4472c51f6dedf626fa89c3cf34]</td>\n",
       "      <td>tcga-ol-a66h-01a</td>\n",
       "      <td>GRCh37:chr3:+:12447485:12447485/C/G</td>\n",
       "      <td>GRCh37:chr3:+:12447485:12447485/C/G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PARP8</td>\n",
       "      <td>GRCh37:chr3:+:12447485:12447485</td>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8029</th>\n",
       "      <td>17592235128885</td>\n",
       "      <td>tcga-ol-a66h-01a-GRCh37:chr8:+:10464643:104646...</td>\n",
       "      <td>[tcga-brca, d584100cdf5d68946a6a47e3254d550e]</td>\n",
       "      <td>tcga-ol-a66h-01a</td>\n",
       "      <td>GRCh37:chr8:+:10464643:10464643/T/A</td>\n",
       "      <td>GRCh37:chr8:+:10464643:10464643/T/A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>ASB9</td>\n",
       "      <td>GRCh37:chr8:+:10464643:10464643</td>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8030</th>\n",
       "      <td>17592235128887</td>\n",
       "      <td>tcga-ol-a66h-01a-GRCh37:chr17:+:19584880:19584...</td>\n",
       "      <td>[tcga-brca, 23aa10c05ed127c8ea6ce17a3b3a6f29]</td>\n",
       "      <td>tcga-ol-a66h-01a</td>\n",
       "      <td>GRCh37:chr17:+:19584880:19584880/A/G</td>\n",
       "      <td>GRCh37:chr17:+:19584880:19584880/A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>ZNF660</td>\n",
       "      <td>GRCh37:chr17:+:19584880:19584880</td>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8031</th>\n",
       "      <td>17592235128889</td>\n",
       "      <td>tcga-ol-a66h-01a-GRCh37:chr4:+:88411538:884115...</td>\n",
       "      <td>[tcga-brca, 985874c25fcacc499fb1b59e6eecf464]</td>\n",
       "      <td>tcga-ol-a66h-01a</td>\n",
       "      <td>GRCh37:chr4:+:88411538:88411538/C/G</td>\n",
       "      <td>GRCh37:chr4:+:88411538:88411538/C/G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SIPA1L2</td>\n",
       "      <td>GRCh37:chr4:+:88411538:88411538</td>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8032</th>\n",
       "      <td>17592235128891</td>\n",
       "      <td>tcga-ol-a66h-01a-GRCh37:chr10:+:63170376:63170...</td>\n",
       "      <td>[tcga-brca, 46d6155733290f7acea12f7a190abec7]</td>\n",
       "      <td>tcga-ol-a66h-01a</td>\n",
       "      <td>GRCh37:chr10:+:63170376:63170376/C/G</td>\n",
       "      <td>GRCh37:chr10:+:63170376:63170376/C/G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PRR33</td>\n",
       "      <td>GRCh37:chr10:+:63170376:63170376</td>\n",
       "      <td>:variant.type/snp</td>\n",
       "      <td>:variant.classification/missense</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5112 rows × 12 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 15
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
